<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# react_perf/jsx-no-new-function-as-prop

<div class="rule-meta">
</div>

### What it does

Prevent Functions that are local to the current method from being used as values of JSX props

### Example

```javascript
// Bad
<Item callback={new Function(...)} />
<Item callback={this.props.callback || function() {}} />

// Good
<Item callback={this.props.callback} />
```
