<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# react_perf/jsx-no-new-array-as-prop

<div class="rule-meta">
</div>

### What it does

Prevent Arrays that are local to the current method from being used as values of JSX props

### Example
```javascript
// Bad
<Item list={[]} />
<Item list={new Array()} />
<Item list={Array()} />
<Item list={this.props.list || []} />
<Item list={this.props.list ? this.props.list : []} />

// Good
<Item list={this.props.list} />
```

