<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# react/jsx-no-script-url <Badge type="info" text="Suspicious" />

<div class="rule-meta">
<Alert class="fix" type="info">
<span class="emoji">ðŸš§</span> An auto-fix is still under development.
</Alert>
</div>

### What it does

Disallow usage of `javascript:` URLs

### Why is this bad?

URLs starting with `javascript:` are a dangerous attack surface because itâ€™s easy to accidentally include unsanitized output in a tag like `<a href>` and create a security hole.
In React 16.9 any URLs starting with `javascript:` scheme log a warning.
In a future major release, React will throw an error if it encounters a `javascript:` URL.

### Examples

Examples of **incorrect** code for this rule:

```jsx
<a href="javascript:void(0)">Test</a>
```

Examples of **correct** code for this rule:

```jsx
<Foo test="javascript:void(0)" />
```

## How to use

To **enable** this rule in the CLI or using the config file, you can use:

::: code-group

```bash [CLI]
oxlint --deny react/jsx-no-script-url
```

```json [Config (.oxlintrc.json)]
{
  "rules": {
    "react/jsx-no-script-url": "error"
  }
}
```

:::

## References

- [Rule Source](https://github.com/oxc-project/oxc/blob/aba3654b81166a7d52f13ac067ffa4bab5e702c9/crates/oxc_linter/src/rules/react/jsx_no_script_url.rs)
