<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# react/jsx-no-script-url <Badge type="info" text="Suspicious" />

<div class="rule-meta">
<Alert class="fix" type="info">
<span class="emoji">ðŸš§</span> An auto-fix is still under development.
</Alert>
</div>

### What it does

Disallow usage of `javascript:` URLs

### Why is this bad?

URLs starting with `javascript:` are a dangerous attack surface because itâ€™s easy to accidentally include unsanitized output in a tag like `<a href>` and create a security hole.
In React 16.9 any URLs starting with `javascript:` scheme log a warning.
In a future major release, React will throw an error if it encounters a `javascript:` URL.

### Examples

Examples of **incorrect** code for this rule:

```jsx
<a href="javascript:void(0)">Test</a>
```

Examples of **correct** code for this rule:

```jsx
<Foo test="javascript:void(0)" />
```

## References

- [Rule Source](https://github.com/oxc-project/oxc/blob/457aa31d92800aaf37a12c695928e8d941308d33/crates/oxc_linter/src/rules/react/jsx_no_script_url.rs)
