<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# react/jsx-no-comment-textnodes <Badge type="info" text="Suspicious" />

<div class="rule-meta">
</div>

### What it does

This rule prevents comment strings (e.g. beginning with `//` or `/*`) from being accidentally injected as a text node in JSX statements.

### Why is this bad?

In JSX, any text node that is not wrapped in curly braces is considered a literal string to be rendered. This can lead to unexpected behavior when the text contains a comment.

### Example

```jsx
// Incorrect:

const Hello = () => {
  return <div>// empty div</div>;
};

const Hello = () => {
  return <div>/* empty div */</div>;
};

// Correct:

const Hello = () => {
  return <div>// empty div</div>;
};

const Hello = () => {
  return <div>{/* empty div */}</div>;
};
```

## How to use

To **enable** this rule in the CLI or using the config file, you can use:

::: code-group

```bash [CLI]
oxlint --deny react/jsx-no-comment-textnodes
```

```json [Config (.oxlintrc.json)]
{
  "rules": {
    "react/jsx-no-comment-textnodes": "error"
  }
}
```

:::

## References

- [Rule Source](https://github.com/oxc-project/oxc/blob/bc0670c8a9937f35f43c431ab616950d8f19f8f7/crates/oxc_linter/src/rules/react/jsx_no_comment_textnodes.rs)
