<!-- This file is auto-generated by tasks/website_linter/src/rules/doc_page.rs. Do not edit it manually. -->

<script setup>
import { data } from '../version.data.js';
const source = `https://github.com/oxc-project/oxc/blob/${ data }/crates/oxc_linter/src/rules/react/no_array_index_key.rs`;
</script>

# react/no-array-index-key <Badge type="info" text="Perf" />

<div class="rule-meta">
</div>

### What it does

Warn if an element uses an Array index in its key.

### Why is this bad?

It's a bad idea to use the array index since it doesn't uniquely identify your elements.
In cases where the array is sorted or an element is added to the beginning of the array,
the index will be changed even though the element representing that index may be the same.
This results in unnecessary renders.

### Examples

Examples of **incorrect** code for this rule:

```jsx
things.map((thing, index) => (
    <Hello key={index} />
));
```

Examples of **correct** code for this rule:

```jsx
things.map((thing, index) => (
    <Hello key={thing.id} />
));
```

## How to use

To **enable** this rule using the config file or in the CLI, you can use:

::: code-group

```json [Config (.oxlintrc.json)]
{
    "plugins": ["react"],
    "rules": {
        "react/no-array-index-key": "error"
    }
}
```

```bash [CLI]
oxlint --deny react/no-array-index-key --react-plugin
```

:::

## References

- <a v-bind:href="source" target="_blank" rel="noreferrer">Rule Source</a>
