<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# react/no-unknown-property

<div class="rule-meta">
</div>

### What it does
Disallow usage of unknown DOM property.

### Why is this bad?
You can use unknown property name that has no effect.

### Example
```jsx
 // Unknown properties
 const Hello = <div class="hello">Hello World</div>;
 const Alphabet = <div abc="something">Alphabet</div>;

 // Invalid aria-* attribute
 const IconButton = <div aria-foo="bar" />;
```

