<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# typescript/consistent-type-definitions <Badge type="info" text="Style" />

<div class="rule-meta">
<Alert class="fix" type="info">
<span class="emoji">üõ†Ô∏è</span> An auto-fix is available for this rule.
</Alert>
</div>

### What it does

Enforce type definitions to consistently use either interface or type.

### Why is this bad?

TypeScript provides two common ways to define an object type: interface and type.
The two are generally very similar, and can often be used interchangeably.
Using the same type declaration style consistently helps with code readability.

### Examples

By default this rule enforces the use of interfaces for object types.

Examples of **incorrect** code for this rule:

```typescript
type T = { x: number };
```

Examples of **correct** code for this rule:

```typescript
type T = string;
type Foo = string | {};

interface T {
  x: number;
}
```

### Options

This rule has a single string option:

`{ type: string, default: "interface" }`

### interface

This is the default option.

### type

Enforces the use of types for object type definitions.

Examples of **incorrect** code for this option:

```typescript
interface T {
  x: number;
}
```

Examples of **correct** code for this option:

```typescript
type T = { x: number };
```

## How to use

To **enable** this rule in the CLI or using the config file, you can use:

::: code-group

```bash [CLI]
oxlint --deny typescript/consistent-type-definitions
```

```json [Config (.oxlintrc.json)]
{
  "rules": {
    "typescript/consistent-type-definitions": "error"
  }
}
```

:::

## References

- [Rule Source](https://github.com/oxc-project/oxc/blob/bc0670c8a9937f35f43c431ab616950d8f19f8f7/crates/oxc_linter/src/rules/typescript/consistent_type_definitions.rs)
