<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

<script setup>
import { data } from '../version.data.js';
const source = `https://github.com/oxc-project/oxc/blob/${ data }/crates/oxc_linter/src/rules/typescript/consistent_indexed_object_style.rs`;
</script>

# typescript/consistent-indexed-object-style <Badge type="info" text="Style" />

<div class="rule-meta">
<Alert class="fix" type="info">
<span class="emoji">üõ†Ô∏è</span> An auto-fix is available for this rule for some violations.
</Alert>
</div>

### What it does

Choose between requiring either `Record` type or indexed signature types.

### Why is this bad?

Inconsistent style for indexed object types can harm readability in a project.

### Examples

Examples of **incorrect** code for this rule with the default "record":

```ts
/*eslint consistent-indexed-object-style: ["error", "record"]*/

interface Foo {
  [key: string]: unknown;
}
type Foo = {
  [key: string]: unknown;
};
```

Examples of **correct** code for this rule:

```ts
/*eslint consistent-indexed-object-style: ["error", "record"]*/

type Foo = Record<string, unknown>;
```

Examples of **incorrect** code for this rule with "index-signature":

```ts
/*eslint consistent-indexed-object-style: ["error", "index-signature"]*/

type Foo = Record<string, unknown>;
```

Examples of **correct** code for this rule:

```ts
/*eslint consistent-indexed-object-style: ["error", "index-signature"]*/

interface Foo {
  [key: string]: unknown;
}
type Foo = {
  [key: string]: unknown;
};
```

## How to use

To **enable** this rule in the CLI or using the config file, you can use:

::: code-group

```bash [CLI]
oxlint --deny typescript/consistent-indexed-object-style
```

```json [Config (.oxlintrc.json)]
{
  "rules": {
    "typescript/consistent-indexed-object-style": "error"
  }
}
```

:::

## References

- <a v-bind:href="source" target="_blank" rel="noreferrer">Rule Source</a>
