<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# typescript/explicit-function-return-type <Badge type="info" text="Restriction" />

<div class="rule-meta">
</div>

### What it does

This rule enforces that functions do have an explicit return type annotation.

### Why is this bad?

Explicit return types do make it visually more clear what type is
returned by a function. They can also speed up TypeScript type checking
performance in large codebases with many large functions.

### Example

Examples of **incorrect** code for this rule:

```ts
// Should indicate that no value is returned (void)
function test() {
  return;
}

// Should indicate that a number is returned
var fn = function () {
  return 1;
};

// Should indicate that a string is returned
var arrowFn = () => "test";

class Test {
  // Should indicate that no value is returned (void)
  method() {
    return;
  }
}
```

Examples of **correct** code for this rule:

```ts
// No return value should be expected (void)
function test(): void {
  return;
}

// A return value of type number
var fn = function (): number {
  return 1;
};

// A return value of type string
var arrowFn = (): string => "test";

class Test {
  // No return value should be expected (void)
  method(): void {
    return;
  }
}
```

## How to use

To **enable** this rule in the CLI or using the config file, you can use:

::: code-group

```bash [CLI]
oxlint --deny typescript/explicit-function-return-type
```

```json [Config (.oxlintrc.json)]
{
  "rules": {
    "typescript/explicit-function-return-type": "error"
  }
}
```

:::

## References

- [Rule Source](https://github.com/oxc-project/oxc/blob/30318457d425dbf627aa428aad8004f6b92b1c59/crates/oxc_linter/src/rules/typescript/explicit_function_return_type.rs)
