<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# typescript/no-non-null-asserted-optional-chain <Badge type="info" text="Correctness" />

<div class="rule-meta">
<Alert class="default-on" type="success">
<span class="emoji">âœ…</span> This rule is turned on by default.
</Alert>
</div>

### What it does

Disallow non-null assertions after an optional chain expression.

### Why is this bad?

`?.` optional chain expressions provide undefined if an object is null or undefined.
Using a `!` non-null assertion to assert the result of an `?.` optional chain expression is non-nullable is likely wrong.

Most of the time, either the object was not nullable and did not need the `?.` for its property lookup, or the `!` is incorrect and introducing a type safety hole.

### Example

```ts
foo?.bar!;
foo?.bar()!;
```

## How to use

To **enable** this rule in the CLI or using the config file, you can use:

::: code-group

```bash [CLI]
oxlint --deny typescript/no-non-null-asserted-optional-chain
```

```json [Config (.oxlintrc.json)]
{
  "rules": {
    "typescript/no-non-null-asserted-optional-chain": "error"
  }
}
```

:::

## References

- [Rule Source](https://github.com/oxc-project/oxc/blob/b9ab60bde696d2742d3c5781084ee3c7bb99821e/crates/oxc_linter/src/rules/typescript/no_non_null_asserted_optional_chain.rs)
