<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# typescript/no-this-alias <Badge type="info" text="Correctness" />

<div class="rule-meta">
<Alert class="default-on" type="success">
<span class="emoji">âœ…</span> This rule is turned on by default.
</Alert>
</div>

### What it does

Disallow unnecessary constraints on generic types.

### Why is this bad?

Generic type parameters (`<T>`) in TypeScript may be "constrained" with
an extends keyword. When no extends is provided, type parameters
default a constraint to unknown. It is therefore redundant to extend
from any or unknown.

the rule doesn't allow `const {allowedName} = this`
this is to keep 1:1 with eslint implementation
sampe with `obj.<allowedName> = this`

````


## How to use

To **enable** this rule in the CLI or using the config file, you can use:

::: code-group

```bash [CLI]
oxlint --deny typescript/no-this-alias
````

```json [Config (.oxlintrc.json)]
{
  "rules": {
    "typescript/no-this-alias": "error"
  }
}
```

:::

## References

- [Rule Source](https://github.com/oxc-project/oxc/blob/b31ab8773865e3f2f05afd8aca9ee541f31a5b90/crates/oxc_linter/src/rules/typescript/no_this_alias.rs)
