<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# typescript/no-this-alias <Badge type="info" text="Correctness" />

<div class="rule-meta">
<Alert class="default-on" type="success">
<span class="emoji">âœ…</span> This rule is turned on by default.
</Alert>
</div>

### What it does

Disallow unnecessary constraints on generic types.

### Why is this bad?

Generic type parameters (`<T>`) in TypeScript may be "constrained" with
an extends keyword. When no extends is provided, type parameters
default a constraint to unknown. It is therefore redundant to extend
from any or unknown.

the rule doesn't allow `const {allowedName} = this`
this is to keep 1:1 with eslint implementation
sampe with `obj.<allowedName> = this`

```


## References
- [Rule Source](https://github.com/oxc-project/oxc/blob/57b7ca8eae8a1d308119ac0e0a8e657e5e61b4bb/crates/oxc_linter/src/rules/typescript/no_this_alias.rs)
```
