<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

<script setup>
import { data } from '../version.data.js';
const source = `https://github.com/oxc-project/oxc/blob/${ data }/crates/oxc_linter/src/rules/typescript/no_unsafe_type_assertion.rs`;
</script>

# typescript/no-unsafe-type-assertion <Badge type="info" text="Suspicious" />

<div class="rule-meta">
<Alert type="info">
<span class="emoji">ðŸ’­</span> This rule requires <a href="https://oxc.rs/docs/guide/usage/linter/type-aware.html" target="_blank" rel="noreferrer">type information</a>.
</Alert>
<Alert class="fix" type="info">
<span class="emoji">ðŸš§</span> An auto-fix is still under development.
</Alert>
</div>

### What it does

This rule disallows type assertions using the `any` type.

### Why is this bad?

Type assertions using `any` completely bypass TypeScript's type system and can lead to runtime errors. They should be avoided in favor of more specific type assertions or proper type guards.

### Examples

Examples of **incorrect** code for this rule:

```ts
declare const value: unknown;

const str = value as any; // unsafe type assertion

const obj = value as any as string; // double assertion through any

function processValue(input: unknown) {
  const processed = input as any; // unsafe
  return processed.someProperty;
}
```

Examples of **correct** code for this rule:

```ts
declare const value: unknown;

// Use specific type assertions
const str = value as string; // more specific assertion

// Use type guards
if (typeof value === "string") {
  const str2 = value; // safe, no assertion needed
}

// Use proper interface assertions
interface User {
  name: string;
  age: number;
}

const user = value as User; // specific type assertion

// Use unknown for truly unknown values
const unknown: unknown = value;
```

## How to use

To **enable** this rule in the CLI or using the config file, you can use:

::: code-group

```bash [CLI]
oxlint --type-aware --deny typescript/no-unsafe-type-assertion
```

```json [Config (.oxlintrc.json)]
{
  "rules": {
    "typescript/no-unsafe-type-assertion": "error"
  }
}
```

:::

## References

- <a v-bind:href="source" target="_blank" rel="noreferrer">Rule Source</a>
