<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# typescript/prefer-for-of

<div class="rule-meta">
</div>

### What it does

Enforces the use of for-of loop instead of a for loop with a simple iteration.

### Why is this bad?

Using a for loop with a simple iteration over an array can be replaced with a more concise
and readable for-of loop. For-of loops are easier to read and less error-prone, as they
eliminate the need for an index variable and manual array access.

### Example

```ts
// Bad
for (let i = 0; i < arr.length; i++) {
  console.log(arr[i]);
}

// Good
for (const item of arr) {
  console.log(item);
}
```
