<!-- This file is auto-generated by tasks/website_linter/src/rules/doc_page.rs. Do not edit it manually. -->

<script setup>
import { data } from '../version.data.js';
const source = `https://github.com/oxc-project/oxc/blob/${ data }/crates/oxc_linter/src/rules/typescript/prefer_includes.rs`;
const tsgolintSource = `https://github.com/oxc-project/tsgolint/blob/main/internal/rules/prefer_includes/prefer_includes.go`;
</script>

# typescript/prefer-includes <Badge type="info" text="Pedantic" />

<div class="rule-meta">
<Alert type="info">
<span class="emoji">ðŸ’­</span> This rule requires <a href="https://oxc.rs/docs/guide/usage/linter/type-aware.html" target="_blank" rel="noreferrer">type information</a>.
</Alert>
<Alert class="fix" type="info">
<span class="emoji">ðŸš§</span> An auto-fix is planned for this rule, but not implemented at this time.
</Alert>
</div>

### What it does

Enforce using `.includes()` instead of `.indexOf() !== -1` or `/regex/.test()`.

### Why is this bad?

`.includes()` is more readable and expressive than checking `.indexOf() !== -1`.
It clearly communicates the intent to check for the presence of a value.
Additionally, for simple string searches, `.includes()` is often preferred over
regex `.test()` for better performance and clarity.

### Examples

Examples of **incorrect** code for this rule:

```ts
// Using indexOf
const str = 'hello world';
if (str.indexOf('world') !== -1) {
  console.log('found');
}

if (str.indexOf('world') != -1) {
  console.log('found');
}

if (str.indexOf('world') > -1) {
  console.log('found');
}

// Using regex test for simple strings
if (/world/.test(str)) {
  console.log('found');
}

// Arrays
const arr = [1, 2, 3];
if (arr.indexOf(2) !== -1) {
  console.log('found');
}
```

Examples of **correct** code for this rule:

```ts
// Using includes for strings
const str = 'hello world';
if (str.includes('world')) {
  console.log('found');
}

// Using includes for arrays
const arr = [1, 2, 3];
if (arr.includes(2)) {
  console.log('found');
}

// Complex regex patterns are allowed
if (/wo+rld/.test(str)) {
  console.log('found');
}

// Regex with flags
if (/world/i.test(str)) {
  console.log('found');
}
```

## How to use

To **enable** this rule using the config file or in the CLI, you can use:

::: code-group

```json [Config (.oxlintrc.json)]
{
    "rules": {
        "typescript/prefer-includes": "error"
    }
}
```

```bash [CLI]
oxlint --type-aware --deny typescript/prefer-includes
```

:::

## References

- <a v-bind:href="source" target="_blank" rel="noreferrer">Rule Source</a>
- <a v-bind:href="tsgolintSource" target="_blank" rel="noreferrer">Rule Source (tsgolint)</a>
