<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# nextjs/no-script-component-in-head <Badge type="info" text="Correctness" />

<div class="rule-meta">
</div>

### What it does

### Why is this bad?

### Example

```javascript

```

## References

- [Rule Source](https://github.com/oxc-project/oxc/blob/5172f324579e6a3858a816912e6d8a284d5debad/crates/oxc_linter/src/rules/nextjs/no_script_component_in_head.rs)
