<!-- This file is auto-generated by tasks/website_linter/src/rules/doc_page.rs. Do not edit it manually. -->

<script setup>
import { data } from '../version.data.js';
const source = `https://github.com/oxc-project/oxc/blob/${ data }/crates/oxc_linter/src/rules/jsdoc/require_returns_description.rs`;
</script>

# jsdoc/require-returns-description <Badge type="info" text="Pedantic" />

<div class="rule-meta">
</div>

### What it does

Requires that the `@returns` tag has a description value.
The error will not be reported if the return value is `void `or `undefined` or if it is `Promise<void>` or `Promise<undefined>`.

### Why is this bad?

A `@returns` tag should have a description value.

### Examples

Examples of **incorrect** code for this rule:

```javascript
/** @returns */
function quux (foo) {}
```

Examples of **correct** code for this rule:

```javascript
/** @returns Foo. */
function quux (foo) {}
```

## How to use

To **enable** this rule using the config file or in the CLI, you can use:

::: code-group

```json [Config (.oxlintrc.json)]
{
    "plugins": ["jsdoc"],
    "rules": {
        "jsdoc/require-returns-description": "error"
    }
}
```

```bash [CLI]
oxlint --deny jsdoc/require-returns-description --jsdoc-plugin
```

:::

## References

- <a v-bind:href="source" target="_blank" rel="noreferrer">Rule Source</a>
