<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# vitest/no-import-node-test

<div class="rule-meta">
<span class="fix">
ğŸ› ï¸ An auto-fix is available for this rule.
</span>
</div>

### What it does

This rule warns when `node:test` is imported (usually accidentally). With `--fix`, it will replace the import with `vitest`.

### Examples

```javascript
// invalid
import { test } from "node:test";
import { expect } from "vitest";

test("foo", () => {
  expect(1).toBe(1);
});
```

```javascript
// valid
import { test, expect } from "vitest";

test("foo", () => {
  expect(1).toBe(1);
});
```
