<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# vitest/prefer-to-be-falsy

<div class="rule-meta">
<Alert class="fix" type="info">
<span class="emoji">üõ†Ô∏è</span> An auto-fix is available for this rule.
</Alert>
</div>

### What it does

This rule warns when `toBe(false)` is used with `expect` or `expectTypeOf`. With `--fix`, it will be replaced with `toBeFalsy()`.

### Examples

```javascript
// bad
expect(foo).toBe(false);
expectTypeOf(foo).toBe(false);

// good
expect(foo).toBeFalsy();
expectTypeOf(foo).toBeFalsy();
```
