<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# vitest/require-local-test-context-for-concurrent-snapshots <Badge type="info" text="Correctness" />

<div class="rule-meta">
</div>

### What it does

The rule is intended to ensure that concurrent snapshot tests are executed within a properly configured local test context.

### Examples

Examples of **incorrect** code for this rule:

```js
test.concurrent("myLogic", () => {
  expect(true).toMatchSnapshot();
});

describe.concurrent("something", () => {
  test("myLogic", () => {
    expect(true).toMatchInlineSnapshot();
  });
});
```

Examples of **correct** code for this rule:

```js
test.concurrent("myLogic", ({ expect }) => {
  expect(true).toMatchSnapshot();
});

test.concurrent("myLogic", (context) => {
  context.expect(true).toMatchSnapshot();
});
```

## References

- [Rule Source](https://github.com/oxc-project/oxc/blob/main/crates/oxc_linter/src/rules/vitest/require_local_test_context_for_concurrent_snapshots.rs)
