<!-- This file is auto-generated by tasks/website_linter/src/rules/doc_page.rs. Do not edit it manually. -->

<script setup>
import { data } from '../version.data.js';
const source = `https://github.com/oxc-project/oxc/blob/${ data }/crates/oxc_linter/src/rules/vue/prefer_import_from_vue.rs`;
</script>

# vue/prefer-import-from-vue <Badge type="info" text="Correctness" />

<div class="rule-meta">
<Alert class="fix" type="info">
<span class="emoji">üõ†Ô∏è</span> An auto-fix is available for this rule.
</Alert>
</div>

### What it does

Enforce import from 'vue' instead of import from '@vue/*'.

### Why is this bad?

Imports from the following modules are almost always wrong. You should import from vue instead.

- `@vue/runtime-dom`
- `@vue/runtime-core`
- `@vue/reactivity`
- `@vue/shared`

### Examples

Examples of **incorrect** code for this rule:

```js
import { ref } from "@vue/reactivity";
import { Component } from "@vue/runtime-core";
import { createApp } from "@vue/runtime-dom";
```

Examples of **correct** code for this rule:

```js
import { Component, createApp, ref } from "vue";
```

## How to use

To **enable** this rule in the CLI or using the config file, you can use:

::: code-group

```bash [CLI]
oxlint --deny vue/prefer-import-from-vue --vue-plugin
```

```json [Config (.oxlintrc.json)]
{
  "plugins": ["vue"],
  "rules": {
    "vue/prefer-import-from-vue": "error"
  }
}
```

:::

## References

- <a v-bind:href="source" target="_blank" rel="noreferrer">Rule Source</a>
