<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

<script setup>
import { data } from '../version.data.js';
const source = `https://github.com/oxc-project/oxc/blob/${ data }/crates/oxc_linter/src/rules/unicorn/consistent_empty_array_spread.rs`;
</script>

# unicorn/consistent-empty-array-spread <Badge type="info" text="Pedantic" />

<div class="rule-meta">
<Alert class="fix" type="info">
<span class="emoji">ðŸ’¡</span> A suggestion is available for this rule.
</Alert>
</div>

### What it does

When spreading a ternary in an array, we can use both [] and '' as fallbacks,
but it's better to have consistent types in both branches.

### Why is this bad?

Having consistent types in both branches makes the code easier to read and understand.

### Examples

Examples of **incorrect** code for this rule:

```javascript
const array = [
  a,
  ...(foo ? [b, c] : ""),
];

const array = [
  a,
  ...(foo ? "bc" : []),
];
```

Examples of **correct** code for this rule:

```javascript
const array = [
  a,
  ...(foo ? [b, c] : []),
];

const array = [
  a,
  ...(foo ? "bc" : ""),
];
```

## How to use

To **enable** this rule in the CLI or using the config file, you can use:

::: code-group

```bash [CLI]
oxlint --deny unicorn/consistent-empty-array-spread
```

```json [Config (.oxlintrc.json)]
{
  "rules": {
    "unicorn/consistent-empty-array-spread": "error"
  }
}
```

:::

## References

- <a v-bind:href="source" target="_blank" rel="noreferrer">Rule Source</a>
