<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# unicorn/consistent-existence-index-check <Badge type="info" text="Style" />

<div class="rule-meta">
<Alert class="fix" type="info">
<span class="emoji">üõ†Ô∏è</span> An auto-fix is available for this rule.
</Alert>
</div>

### What it does

Enforce consistent style for element existence checks with `indexOf()`,
`lastIndexOf()`, `findIndex()`, and `findLastIndex()`. This rule ensures
that comparisons for element presence are made with `-1` rather than other
comparison operators like `< 0` or `>= 0`, improving clarity and consistency.

### Why is this bad?

Using `< 0` or `>= 0` for element existence checks can lead to confusion,
especially for developers who are not familiar with the specific behavior of
these methods. The explicit `=== -1` or `!== -1` makes the intent clearer and
more readable, ensuring consistency across the codebase.

### Examples

Examples of **incorrect** code for this rule:

```javascript
const index = foo.indexOf("bar");
if (index < 0) {
}

const index = foo.indexOf("bar");
if (index >= 0) {
}
```

Examples of **correct** code for this rule:

```javascript
const index = foo.indexOf("bar");
if (index === -1) {
}

const index = foo.indexOf("bar");
if (index !== -1) {
}
```

## How to use

To **enable** this rule in the CLI or using the config file, you can use:

::: code-group

```bash [CLI]
oxlint --deny unicorn/consistent-existence-index-check
```

```json [Config (.oxlintrc.json)]
{
  "rules": {
    "unicorn/consistent-existence-index-check": "error"
  }
}
```

:::

## References

- [Rule Source](https://github.com/oxc-project/oxc/blob/30318457d425dbf627aa428aad8004f6b92b1c59/crates/oxc_linter/src/rules/unicorn/consistent_existence_index_check.rs)
