<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# unicorn/filename-case <Badge type="info" text="Style" />

<div class="rule-meta">
</div>

### What it does

Enforces specific case styles for filenames. By default, kebab case is enforced.

### Why is this bad?

Inconsistent file naming conventions can make it harder to locate files or to create new ones.

### Cases

Examples of **correct** filenames for each case:

#### `kebabCase`

- `some-file-name.js`
- `some-file-name.test.js`
- `some-file-name.test-utils.js`

#### `camelCase`

- `someFileName.js`
- `someFileName.test.js`
- `someFileName.testUtils.js`

#### `snakeCase`

- `some_file_name.js`
- `some_file_name.test.js`
- `some_file_name.test_utils.js`

#### `pascalCase`

- `SomeFileName.js`
- `SomeFileName.Test.js`
- `SomeFileName.TestUtils.js`

## How to use

To **enable** this rule in the CLI or using the config file, you can use:

::: code-group

```bash [CLI]
oxlint --deny unicorn/filename-case
```

```json [Config (.oxlintrc.json)]
{
  "rules": {
    "unicorn/filename-case": "error"
  }
}
```

:::

## References

- [Rule Source](https://github.com/oxc-project/oxc/blob/19c4835a02d596d931670721daf996bff74fcbbd/crates/oxc_linter/src/rules/unicorn/filename_case.rs)
