<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# unicorn/no-accessor-recursion <Badge type="info" text="Suspicious" />

<div class="rule-meta">
</div>

### What it does

Disallow recursive access to this within getters and setters

### Why is this bad?

This rule prevents recursive access to this within getter and setter methods in objects and classes,
avoiding infinite recursion and stack overflow errors.

### Examples

Examples of **incorrect** code for this rule:

```js
const foo = {
  get bar() {
    return this.bar;
  },
};
```

Examples of **correct** code for this rule:

```js
const foo = {
  get bar() {
    return this.baz;
  },
};
```

## How to use

To **enable** this rule in the CLI or using the config file, you can use:

::: code-group

```bash [CLI]
oxlint --deny unicorn/no-accessor-recursion
```

```json [Config (.oxlintrc.json)]
{
  "rules": {
    "unicorn/no-accessor-recursion": "error"
  }
}
```

:::

## References

- [Rule Source](https://github.com/oxc-project/oxc/blob/4b4b09e0e44b1e904870920b3dbd2c75398ed9fd/crates/oxc_linter/src/rules/unicorn/no_accessor_recursion.rs)
