<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# unicorn/no-abusive-eslint-disable <Badge type="info" text="Restriction" />

<div class="rule-meta">
</div>

### What it does

This rule disallows `eslint-disable` comments that do not specify any rules to disable.

### Why is this bad?

When only one rule should be disabled but the `eslint-disable` comment does not specify any rules, other useful errors will also be silently ignored.

### Examples

Examples of **incorrect** code for this rule:

```javascript
/* eslint-disable */
console.log(message);

console.log(message); // eslint-disable-line

// eslint-disable-next-line
console.log(message);
```

Examples of **correct** code for this rule:

```javascript
/* eslint-disable no-console */
console.log(message);

console.log(message); // eslint-disable-line no-console

// eslint-disable-next-line no-console
console.log(message);
```

## References

- [Rule Source](https://github.com/oxc-project/oxc/blob/5172f324579e6a3858a816912e6d8a284d5debad/crates/oxc_linter/src/rules/unicorn/no_abusive_eslint_disable.rs)
