<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

<script setup>
import { data } from '../version.data.js';
const source = `https://github.com/oxc-project/oxc/blob/${ data }/crates/oxc_linter/src/rules/unicorn/no_array_sort.rs`;
</script>

# unicorn/no-array-sort <Badge type="info" text="Suspicious" />

<div class="rule-meta">
<Alert class="fix" type="info">
<span class="emoji">üõ†Ô∏è</span> An auto-fix is available for this rule.
</Alert>
</div>

### What it does

Prefer using `Array#toSorted()` over `Array#sort()`.

### Why is this bad?

`Array#sort()` modifies the original array in place, which can lead to unintended side effects‚Äîespecially
when the original array is used elsewhere in the code.

### Examples

Examples of **incorrect** code for this rule:

```js
const sorted = [...array].sort();
```

Examples of **correct** code for this rule:

```js
const sorted = [...array].toSorted();
```

### Options

#### allowExpressionStatement

`{ type: boolean, default: true }`

This rule allows `array.sort()` as an expression statement by default,
Pass allowExpressionStatement: false to forbid `Array#sort()` even if it's an expression statement.

Examples of **incorrect** code for this rule with the `{ "allowExpressionStatement": false }` option:

```js
array.sort();
```

## How to use

To **enable** this rule in the CLI or using the config file, you can use:

::: code-group

```bash [CLI]
oxlint --deny unicorn/no-array-sort
```

```json [Config (.oxlintrc.json)]
{
  "rules": {
    "unicorn/no-array-sort": "error"
  }
}
```

:::

## References

- <a v-bind:href="source" target="_blank" rel="noreferrer">Rule Source</a>
