<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

<script setup>
import { data } from '../version.data.js';
const source = `https://github.com/oxc-project/oxc/blob/${ data }/crates/oxc_linter/src/rules/unicorn/no_accessor_recursion.rs`;
</script>

# unicorn/no-accessor-recursion <Badge type="info" text="Suspicious" />

<div class="rule-meta">
</div>

### What it does

Disallow recursive access to this within getters and setters

### Why is this bad?

This rule prevents recursive access to this within getter and setter methods in objects and classes,
avoiding infinite recursion and stack overflow errors.

### Examples

Examples of **incorrect** code for this rule:

```js
const foo = {
  get bar() {
    return this.bar;
  },
};
```

Examples of **correct** code for this rule:

```js
const foo = {
  get bar() {
    return this.baz;
  },
};
```

## How to use

To **enable** this rule in the CLI or using the config file, you can use:

::: code-group

```bash [CLI]
oxlint --deny unicorn/no-accessor-recursion
```

```json [Config (.oxlintrc.json)]
{
  "rules": {
    "unicorn/no-accessor-recursion": "error"
  }
}
```

:::

## References

- <a v-bind:href="source" target="_blank" rel="noreferrer">Rule Source</a>
