<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

<script setup>
import { data } from '../version.data.js';
const source = `https://github.com/oxc-project/oxc/blob/${ data }/crates/oxc_linter/src/rules/unicorn/no_empty_file.rs`;
</script>

# unicorn/no-empty-file <Badge type="info" text="Correctness" />

<div class="rule-meta">
<Alert class="default-on" type="success">
<span class="emoji">âœ…</span> This rule is turned on by default.
</Alert>
</div>

### What it does

Disallows any files only containing the following:

- Whitespace
- Comments
- Directives
- Empty statements
- Empty blocks
- Hashbang

### Why is this bad?

Meaningless files clutter a codebase.

## How to use

To **enable** this rule in the CLI or using the config file, you can use:

::: code-group

```bash [CLI]
oxlint --deny unicorn/no-empty-file
```

```json [Config (.oxlintrc.json)]
{
  "rules": {
    "unicorn/no-empty-file": "error"
  }
}
```

:::

## References

- <a v-bind:href="source" target="_blank" rel="noreferrer">Rule Source</a>
