<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

<script setup>
import { data } from '../version.data.js';
const source = `https://github.com/oxc-project/oxc/blob/${ data }/crates/oxc_linter/src/rules/unicorn/no_for_loop.rs`;
</script>

# unicorn/no-for-loop <Badge type="info" text="Style" />

<div class="rule-meta">
<Alert class="fix" type="info">
<span class="emoji">ðŸš§</span> An auto-fix is still under development.
</Alert>
</div>

### What it does

Enforces the use of `for-of` loops over classical counter based `for` loops.

### Why is this bad?

- A `for-of` loop makes intent explicit â€“ _iterate over all elements_.
- It removes boiler-plate (`i`, `arr.length`, manual `++`).

### Examples

Examples of **incorrect** code for this rule:

```js
for (let i = 0; i < items.length; i++) {
  const item = items[i];
  console.log(item);
}
```

Examples of **correct** code for this rule:

```js
for (const item of items) {
  console.log(item);
}
```

## How to use

To **enable** this rule in the CLI or using the config file, you can use:

::: code-group

```bash [CLI]
oxlint --deny unicorn/no-for-loop
```

```json [Config (.oxlintrc.json)]
{
  "rules": {
    "unicorn/no-for-loop": "error"
  }
}
```

:::

## References

- <a v-bind:href="source" target="_blank" rel="noreferrer">Rule Source</a>
