<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

<script setup>
import { data } from '../version.data.js';
const source = `https://github.com/oxc-project/oxc/blob/${ data }/crates/oxc_linter/src/rules/unicorn/no_instanceof_builtins.rs`;
</script>

# unicorn/no-instanceof-builtins <Badge type="info" text="Suspicious" />

<div class="rule-meta">
<Alert class="fix" type="info">
<span class="emoji">ðŸš§</span> An auto-fix is still under development.
</Alert>
</div>

### What it does

Disallows the use of `instanceof` with ECMAScript built-in constructors because:

- it breaks across execution contexts (`iframe`, Web Worker, Node VM, etc.);
- it is often misleading (e.g. `instanceof Array` fails for a subclass);
- there is always a clearer and safer alternative (`Array.isArray`, `typeof`, `Buffer.isBuffer`, â€¦).

### Why is this bad?

`instanceof` breaks across execution contexts (`iframe`, Web Worker, Node `vm`),
and may give misleading results for subclasses or exotic objects.

### Examples

Examples of **incorrect** code for this rule:

```javascript
if (arr instanceof Array) { â€¦ }
if (el instanceof HTMLElement) { â€¦ }
```

Examples of **correct** code for this rule:

```javascript
if (Array.isArray(arr)) { â€¦ }
if (el?.nodeType === 1) { â€¦ }
```

## How to use

To **enable** this rule in the CLI or using the config file, you can use:

::: code-group

```bash [CLI]
oxlint --deny unicorn/no-instanceof-builtins
```

```json [Config (.oxlintrc.json)]
{
  "rules": {
    "unicorn/no-instanceof-builtins": "error"
  }
}
```

:::

## References

- <a v-bind:href="source" target="_blank" rel="noreferrer">Rule Source</a>
