<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

<script setup>
import { data } from '../version.data.js';
const source = `https://github.com/oxc-project/oxc/blob/${ data }/crates/oxc_linter/src/rules/unicorn/no_process_exit.rs`;
</script>

# unicorn/no-process-exit <Badge type="info" text="Restriction" />

<div class="rule-meta">
<Alert class="fix" type="info">
<span class="emoji">ðŸš§</span> An auto-fix is still under development.
</Alert>
</div>

### What it does

Disallow `process.exit()`.

### Why is this bad?

Only use `process.exit()` in CLI apps. Throw an error instead.

### Examples

Examples of **incorrect** code for this rule:

```javascript
if (problem) process.exit(1);
```

Examples of **correct** code for this rule:

```javascript
if (problem) throw new Error("message");
```

```
#!/usr/bin/env node
if (problem) process.exit(1);
```

## How to use

To **enable** this rule in the CLI or using the config file, you can use:

::: code-group

```bash [CLI]
oxlint --deny unicorn/no-process-exit
```

```json [Config (.oxlintrc.json)]
{
  "rules": {
    "unicorn/no-process-exit": "error"
  }
}
```

:::

## References

- <a v-bind:href="source" target="_blank" rel="noreferrer">Rule Source</a>
