---
title: "unicorn/no-useless-length-check"
category: "Correctness"
default: true
type_aware: false
fix: "pending"
---

<!-- This file is auto-generated by tasks/website_linter/src/rules/doc_page.rs. Do not edit it manually. -->

<script setup>
import { data } from '../version.data.js';
const source = `https://github.com/oxc-project/oxc/blob/${ data }/crates/oxc_linter/src/rules/unicorn/no_useless_length_check.rs`;
</script>

<RuleHeader />

### What it does

It checks for an unnecessary array length check in a logical expression.

The cases are:

- `array.length === 0 || array.every(Boolean)` (`array.every` returns `true` if array is has elements)
- `array.length > 0 && array.some(Boolean)` (`array.some` returns `false` if array is empty)

### Why is this bad?

An extra unnecessary length check is done.

### Examples

Examples of **incorrect** code for this rule:

```javascript
if (array.length === 0 || array.every(Boolean)) {
  // do something!
}
```

Examples of **correct** code for this rule:

```javascript
if (array.every(Boolean)) {
  // do something!
}
```

## How to use

To **enable** this rule using the config file or in the CLI, you can use:

::: code-group

```json [Config (.oxlintrc.json)]
{
  "rules": {
    "unicorn/no-useless-length-check": "error"
  }
}
```

```bash [CLI]
oxlint --deny unicorn/no-useless-length-check
```

:::

## References

- <a v-bind:href="source" target="_blank" rel="noreferrer">Rule Source</a>
