---
title: "unicorn/no-unreadable-array-destructuring"
category: "Style"
default: false
type_aware: false
fix: "none"
---

<!-- This file is auto-generated by tasks/website_linter/src/rules/doc_page.rs. Do not edit it manually. -->

<script setup>
import { data } from '../version.data.js';
const source = `https://github.com/oxc-project/oxc/blob/${ data }/crates/oxc_linter/src/rules/unicorn/no_unreadable_array_destructuring.rs`;
</script>

<RuleHeader />

### What it does

Disallow unreadable array destructuring

### Why is this bad?

Destructuring is very useful, but it can also make some code harder to read.
This rule prevents ignoring consecutive values when destructuring from an array.

### Examples

Examples of **incorrect** code for this rule:

```javascript
const [, , foo] = parts;
```

Examples of **correct** code for this rule:

```javascript
const [foo] = parts;
```

## How to use

To **enable** this rule using the config file or in the CLI, you can use:

::: code-group

```json [Config (.oxlintrc.json)]
{
  "rules": {
    "unicorn/no-unreadable-array-destructuring": "error"
  }
}
```

```bash [CLI]
oxlint --deny unicorn/no-unreadable-array-destructuring
```

:::

## References

- <a v-bind:href="source" target="_blank" rel="noreferrer">Rule Source</a>
