<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# unicorn/no-unreadable-array-destructuring <Badge type="info" text="Style" />

<div class="rule-meta">
</div>

### What it does

Disallow unreadable array destructuring

### Why is this bad?

Destructuring is very useful, but it can also make some code harder to read.
This rule prevents ignoring consecutive values when destructuring from an array.

### Example

Examples of **incorrect** code for this rule:

```javascript
const [, , foo] = parts;
```

Examples of **correct** code for this rule:

```javascript
const [foo] = parts;
```

## How to use

To **enable** this rule in the CLI or using the config file, you can use:

::: code-group

```bash [CLI]
oxlint --deny unicorn/no-unreadable-array-destructuring
```

```json [Config (.oxlintrc.json)]
{
  "rules": {
    "unicorn/no-unreadable-array-destructuring": "error"
  }
}
```

:::

## References

- [Rule Source](https://github.com/oxc-project/oxc/blob/19fdf8993df7b697b99d9b92a3a546cce7171c42/crates/oxc_linter/src/rules/unicorn/no_unreadable_array_destructuring.rs)
