<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# unicorn/no-useless-undefined

<div class="rule-meta">
<span class="fix">
ğŸ› ï¸ An auto-fix is available for this rule.
</span>
</div>

### What it does

Do not use useless `undefined`.

### Why is this bad?

`undefined` is the default value for new variables, parameters, return statements, etcâ€¦ so specifying it doesn't make any difference.

### Example

```javascript
// bad
let foo = undefined;
// good:
let foo;
```
