<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# unicorn/no-zero-fractions

<div class="rule-meta">
<span class="fix">
üõ†Ô∏è An auto-fix is available for this rule.
</span>
</div>

### What it does

Prevents the use of zero fractions.

### Why is this bad?

There is no difference in JavaScript between, for example, `1`, `1.0` and `1.`, so prefer the former for consistency and brevity.

### Example

```javascript
// Bad
const foo = 1.0;
const foo = -1.0;
const foo = 123_456.000_000;

// Good
const foo = 1;
const foo = -1;
const foo = 123456;
const foo = 1.1;
```
