<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# unicorn/prefer-date-now

<div class="rule-meta">
<span class="fix">
ðŸš§ An auto-fix is still under development.
</span>
</div>

### What it does

Prefers use of `Date.now()` over `new Date().getTime()` or `new Date().valueOf()`.

### Why is this bad?

Using `Date.now()` is shorter and nicer than `new Date().getTime()`, and avoids unnecessary instantiation of `Date` objects.

### Example

```javascript
// bad
const ts = new Date().getTime();
const ts = new Date().valueOf();

// good
const ts = Date.now();
```
