<!-- This file is auto-generated by tasks/website_linter/src/rules/doc_page.rs. Do not edit it manually. -->

<script setup>
import { data } from '../version.data.js';
const source = `https://github.com/oxc-project/oxc/blob/${ data }/crates/oxc_linter/src/rules/unicorn/prefer_global_this.rs`;
</script>

# unicorn/prefer-global-this <Badge type="info" text="Style" />

<div class="rule-meta">
<Alert class="fix" type="info">
<span class="emoji">ðŸš§</span> An auto-fix is still under development.
</Alert>
</div>

### What it does

Enforces the use of [`globalThis`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/globalThis) instead of
environmentâ€‘specific global object aliases (`window`, `self`, or `global`).
Using the standard `globalThis` makes your code portable across browsers, Webâ€¯Workers, Node.js,
and future JavaScript runtimes.

### Why is this bad?

â€¢ **Portability**Â â€“Â `window` is only defined in browser main threads, `self` is used in Webâ€¯Workers,
and `global` is Nodeâ€‘specific. Choosing the wrong alias causes runtime crashes when the code is
executed outside of its original environment.
â€¢ **Clarity**Â â€“Â `globalThis` clearly communicates that you are referring to the global object itself
rather than a particular platform.

### Examples

Examples of **incorrect** code for this rule:

```js
// Browserâ€‘only
window.alert("Hi");

// Nodeâ€‘only
if (typeof global.Buffer !== "undefined") {}

// Webâ€¯Workerâ€‘only
self.postMessage("done");
```

Examples of **correct** code for this rule:

```js
globalThis.alert("Hi");

if (typeof globalThis.Buffer !== "undefined") {}

globalThis.postMessage("done");
```

## How to use

To **enable** this rule in the CLI or using the config file, you can use:

::: code-group

```bash [CLI]
oxlint --deny unicorn/prefer-global-this
```

```json [Config (.oxlintrc.json)]
{
  "rules": {
    "unicorn/prefer-global-this": "error"
  }
}
```

:::

## References

- <a v-bind:href="source" target="_blank" rel="noreferrer">Rule Source</a>
