<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# unicorn/prefer-reflect-apply <Badge type="info" text="Style" />

<div class="rule-meta">
</div>

### What it does

### Why is this bad?

Reflect.apply() is arguably less verbose and easier to understand.
In addition, when you accept arbitrary methods,
it's not safe to assume .apply() exists or is not overridden.

### Example

Examples of **incorrect** code for this rule:

```javascript
foo.apply(null, [42]);
```

Examples of **correct** code for this rule:

```javascript
Reflect.apply(foo, null);
```

## How to use

To **enable** this rule in the CLI or using the config file, you can use:

::: code-group

```bash [CLI]
oxlint --deny unicorn/prefer-reflect-apply
```

```json [Config (.oxlintrc.json)]
{
  "rules": {
    "unicorn/prefer-reflect-apply": "error"
  }
}
```

:::

## References

- [Rule Source](https://github.com/oxc-project/oxc/blob/b31ab8773865e3f2f05afd8aca9ee541f31a5b90/crates/oxc_linter/src/rules/unicorn/prefer_reflect_apply.rs)
