---
title: "unicorn/prefer-negative-index"
category: "Style"
default: false
type_aware: false
fix: "fixable_fix"
---

<!-- This file is auto-generated by tasks/website_linter/src/rules/doc_page.rs. Do not edit it manually. -->

<script setup>
import { data } from '../version.data.js';
const source = `https://github.com/oxc-project/oxc/blob/${ data }/crates/oxc_linter/src/rules/unicorn/prefer_negative_index.rs`;
</script>

<RuleHeader />

### What it does

Prefer negative index over `.length` - index when possible

### Why is this bad?

Conciseness and readability

### Examples

Examples of **incorrect** code for this rule:

```js
foo.slice(foo.length - 2, foo.length - 1);
foo.at(foo.length - 1);
```

Examples of **correct** code for this rule:

```js
foo.slice(-2, -1);
foo.at(-1);
```

## How to use

To **enable** this rule using the config file or in the CLI, you can use:

::: code-group

```json [Config (.oxlintrc.json)]
{
  "rules": {
    "unicorn/prefer-negative-index": "error"
  }
}
```

```bash [CLI]
oxlint --deny unicorn/prefer-negative-index
```

:::

## References

- <a v-bind:href="source" target="_blank" rel="noreferrer">Rule Source</a>
