<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# unicorn/prefer-dom-node-dataset

<div class="rule-meta">
</div>

### What it does

Use [`.dataset`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/dataset) on DOM elements over `getAttribute(â€¦)`, `.setAttribute(â€¦)`, `.removeAttribute(â€¦)` and `.hasAttribute(â€¦)`.

### Why is this bad?

The `dataset` property is a map of strings that contains all the `data-*` attributes from the element. It is a convenient way to access all of them at once.

### Example

```javascript
// Bad
element.setAttribute("data-unicorn", "ðŸ¦„");

// Good
element.dataset.unicorn = "ðŸ¦„";
```
