<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# unicorn/prefer-dom-node-text-content

<div class="rule-meta">
<Alert class="fix" type="info">
<span class="emoji">üõ†Ô∏è</span> An auto-fix is available for this rule for some violations.
</Alert>
</div>

### What it does

Enforces the use of `.textContent` over `.innerText` for DOM nodes.

### Why is this bad?

There are some disadvantages of using .innerText.

- `.innerText` is much more performance-heavy as it requires layout information to return the result.
- `.innerText` is defined only for HTMLElement objects, while `.textContent` is defined for all Node objects.
- `.innerText` is not standard, for example, it is not present in Firefox.

### Example

```javascript
// Bad
const text = foo.innerText;

// Good
const text = foo.textContent;
```
