<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# unicorn/prefer-logical-operator-over-ternary

<div class="rule-meta">
</div>

### What it does

This rule finds ternary expressions that can be simplified to a logical operator.

### Why is this bad?

Using a logical operator is shorter and simpler than a ternary expression.

### Example
```javascript

// Bad
const foo = bar ? bar : baz;
console.log(foo ? foo : bar);

// Good
const foo = bar || baz;
console.log(foo ?? bar);

```

