<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# unicorn/prefer-dom-node-dataset <Badge type="info" text="Pedantic" />

<div class="rule-meta">
<Alert class="fix" type="info">
<span class="emoji">ðŸš§</span> An auto-fix is still under development.
</Alert>
</div>

### What it does

Use [`.dataset`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/dataset) on DOM elements over `getAttribute(â€¦)`, `.setAttribute(â€¦)`, `.removeAttribute(â€¦)` and `.hasAttribute(â€¦)`.

### Why is this bad?

The `dataset` property is a map of strings that contains all the `data-*` attributes from the element. It is a convenient way to access all of them at once.

### Example

Examples of **incorrect** code for this rule:

```javascript
element.setAttribute("data-unicorn", "ðŸ¦„");
```

Examples of **correct** code for this rule:

```javascript
element.dataset.unicorn = "ðŸ¦„";
```

## How to use

To **enable** this rule in the CLI or using the config file, you can use:

::: code-group

```bash [CLI]
oxlint --deny unicorn/prefer-dom-node-dataset
```

```json [Config (.oxlintrc.json)]
{
  "rules": {
    "unicorn/prefer-dom-node-dataset": "error"
  }
}
```

:::

## References

- [Rule Source](https://github.com/oxc-project/oxc/blob/30318457d425dbf627aa428aad8004f6b92b1c59/crates/oxc_linter/src/rules/unicorn/prefer_dom_node_dataset.rs)
