<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# unicorn/prefer-modern-dom-apis <Badge type="info" text="Style" />

<div class="rule-meta">
<Alert class="fix" type="info">
<span class="emoji">ðŸš§</span> An auto-fix is still under development.
</Alert>
</div>

### What it does

Enforces the use of:

- `childNode.replaceWith(newNode)` over `parentNode.replaceChild(newNode, oldNode)`
- `referenceNode.before(newNode)` over `parentNode.insertBefore(newNode, referenceNode)`
- `referenceNode.before('text')` over `referenceNode.insertAdjacentText('beforebegin', 'text')`
- `referenceNode.before(newNode)` over `referenceNode.insertAdjacentElement('beforebegin', newNode)`

### Why is this bad?

There are some advantages of using the newer DOM APIs, like:

- Traversing to the parent node is not necessary.
- Appending multiple nodes at once.
- Both `DOMString` and DOM node objects can be manipulated.

### Example

Examples of **incorrect** code for this rule:

```javascript
oldChildNode.replaceWith(newChildNode);
```

Examples of **correct** code for this rule:

```javascript
parentNode.replaceChild(newChildNode, oldChildNode);
```

## References

- [Rule Source](https://github.com/oxc-project/oxc/blob/4bb1dca6730cf630cc8c53db2bfd03f2d0f3dd8f/crates/oxc_linter/src/rules/unicorn/prefer_modern_dom_apis.rs)
