---
title: "unicorn/prefer-modern-dom-apis"
category: "Style"
default: false
type_aware: false
fix: "pending"
---

<!-- This file is auto-generated by tasks/website_linter/src/rules/doc_page.rs. Do not edit it manually. -->

<script setup>
import { data } from '../version.data.js';
const source = `https://github.com/oxc-project/oxc/blob/${ data }/crates/oxc_linter/src/rules/unicorn/prefer_modern_dom_apis.rs`;
</script>

<RuleHeader />

### What it does

Enforces the use of:

- `childNode.replaceWith(newNode)` over `parentNode.replaceChild(newNode, oldNode)`
- `referenceNode.before(newNode)` over `parentNode.insertBefore(newNode, referenceNode)`
- `referenceNode.before('text')` over `referenceNode.insertAdjacentText('beforebegin', 'text')`
- `referenceNode.before(newNode)` over `referenceNode.insertAdjacentElement('beforebegin', newNode)`

### Why is this bad?

There are some advantages of using the newer DOM APIs, like:

- Traversing to the parent node is not necessary.
- Appending multiple nodes at once.
- Both `DOMString` and DOM node objects can be manipulated.

### Examples

Examples of **incorrect** code for this rule:

```javascript
oldChildNode.replaceWith(newChildNode);
```

Examples of **correct** code for this rule:

```javascript
parentNode.replaceChild(newChildNode, oldChildNode);
```

## How to use

To **enable** this rule using the config file or in the CLI, you can use:

::: code-group

```json [Config (.oxlintrc.json)]
{
  "rules": {
    "unicorn/prefer-modern-dom-apis": "error"
  }
}
```

```bash [CLI]
oxlint --deny unicorn/prefer-modern-dom-apis
```

:::

## References

- <a v-bind:href="source" target="_blank" rel="noreferrer">Rule Source</a>
