<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# unicorn/text-encoding-identifier-case

<div class="rule-meta">
<Alert class="fix" type="info">
<span class="emoji">ðŸš§</span> An auto-fix is still under development.
</Alert>
</div>

### What it does

This rule aims to enforce consistent case for text encoding identifiers.

Enforces `'utf8'` for UTF-8 encoding
Enforces `'ascii'` for ASCII encoding.

### Example

```javascript
import fs from "node:fs/promises";
async function bad() {
  await fs.readFile(file, "UTF-8");

  await fs.readFile(file, "ASCII");

  const string = buffer.toString("utf-8");
}

async function good() {
  await fs.readFile(file, "utf8");

  await fs.readFile(file, "ascii");

  const string = buffer.toString("utf8");
}
```
