<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# unicorn/text-encoding-identifier-case <Badge type="info" text="Style" />

<div class="rule-meta">
<Alert class="fix" type="info">
<span class="emoji">üõ†Ô∏è</span> An auto-fix is available for this rule.
</Alert>
</div>

### What it does

This rule aims to enforce consistent case for text encoding identifiers.

Enforces `'utf8'` for UTF-8 encoding
Enforces `'ascii'` for ASCII encoding.

### Why is this bad?

- Inconsistency in text encoding identifiers can make the code harder to read and understand.
- The ECMAScript specification does not define the case sensitivity of text encoding identifiers, but it is common practice to use lowercase.

### Example

Examples of **incorrect** code for this rule:

```javascript
import fs from "node:fs/promises";
async function bad() {
  await fs.readFile(file, "UTF-8");

  await fs.readFile(file, "ASCII");

  const string = buffer.toString("utf-8");
}
```

Examples of **correct** code for this rule:

```javascript
async function good() {
  await fs.readFile(file, "utf8");

  await fs.readFile(file, "ascii");

  const string = buffer.toString("utf8");
}
```

## How to use

To **enable** this rule in the CLI or using the config file, you can use:

::: code-group

```bash [CLI]
oxlint --deny unicorn/text-encoding-identifier-case
```

```json [Config (.oxlintrc.json)]
{
  "rules": {
    "unicorn/text-encoding-identifier-case": "error"
  }
}
```

:::

## References

- [Rule Source](https://github.com/oxc-project/oxc/blob/0f1e0e87715075c250763ea31c3a82505a4f10d3/crates/oxc_linter/src/rules/unicorn/text_encoding_identifier_case.rs)
