<!-- This file is auto-generated by tasks/website_linter/src/rules/doc_page.rs. Do not edit it manually. -->

<script setup>
import { data } from '../version.data.js';
const source = `https://github.com/oxc-project/oxc/blob/${ data }/crates/oxc_linter/src/rules/import/extensions.rs`;
</script>

# import/extensions <Badge type="info" text="Restriction" />

<div class="rule-meta">
</div>

### What it does

Some file resolve algorithms allow you to omit the file extension within the import source path.
For example the node resolver (which does not yet support ESM/import) can resolve ./foo/bar to the absolute path /User/someone/foo/bar.js because the .js extension is resolved automatically by default in CJS.
Depending on the resolver you can configure more extensions to get resolved automatically.
In order to provide a consistent use of file extensions across your code base, this rule can enforce or disallow the use of certain file extensions.

### Why is this bad?

ESM-based file resolve algorithms (e.g., the one that Vite provides) recommend specifying the file extension to improve performance.

### Examples

Examples of **incorrect** code for this rule:

The following patterns are considered problems when configuration set to "always":

```js
import foo from "@/foo";
import bar from "./bar";
import Component from "./Component";
import foo from "./foo";
```

The following patterns are considered problems when configuration set to "never":

```js
import express from "express/index.js";
import bar from "./bar.json";
import Component from "./Component.jsx";
import foo from "./foo.js";
```

Examples of **correct** code for this rule:

The following patterns are not considered problems when configuration set to "always":

```js
import foo from "@/foo.js";
import * as path from "path";
import bar from "./bar.json";
import Component from "./Component.jsx";
import foo from "./foo.js";
```

The following patterns are not considered problems when configuration set to "never":

```js
import express from "express/index";
import * as path from "path";
import bar from "./bar";
import Component from "./Component";
import foo from "./foo";
```

## Configuration

This rule accepts a configuration object with the following properties:

### checkTypeImports

type: `boolean`

default: `false`

Whether to check type imports when enforcing extension rules.

### ignorePackages

type: `boolean`

default: `true`

Whether to ignore package imports (e.g., 'react', 'lodash') when enforcing extension rules.

### js

type: `"always" | "never" | "ignorePackages"`

default: `"never"`

Configuration for `.js` file extensions.

### json

type: `"always" | "never" | "ignorePackages"`

default: `"never"`

Configuration for `.json` file extensions.

### jsx

type: `"always" | "never" | "ignorePackages"`

default: `"never"`

Configuration for `.jsx` file extensions.

## requireExtension

type: `string | null`

default: `null`

Configuration for requiring or disallowing file extensions in import/require statements.

### ts

type: `"always" | "never" | "ignorePackages"`

default: `"never"`

Configuration for `.ts` file extensions.

### tsx

type: `"always" | "never" | "ignorePackages"`

default: `"never"`

Configuration for `.tsx` file extensions.

## How to use

To **enable** this rule in the CLI or using the config file, you can use:

::: code-group

```bash [CLI]
oxlint --deny import/extensions --import-plugin
```

```json [Config (.oxlintrc.json)]
{
  "plugins": ["import"],
  "rules": {
    "import/extensions": "error"
  }
}
```

:::

## References

- <a v-bind:href="source" target="_blank" rel="noreferrer">Rule Source</a>
