<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

<script setup>
import { data } from '../version.data.js';
const source = `https://github.com/oxc-project/oxc/blob/${ data }/crates/oxc_linter/src/rules/import/max_dependencies.rs`;
</script>

# import/max-dependencies <Badge type="info" text="Pedantic" />

<div class="rule-meta">
</div>

### What it does

Forbid modules to have too many dependencies (import or require statements).

### Why is this bad?

This is a useful rule because a module with too many dependencies is a code smell,
and usually indicates the module is doing too much and/or should be broken up into
smaller modules.

### Examples

Given `{"max": 2}`

Examples of **incorrect** code for this rule:

```javascript
import a from "./a";
import b from "./b";
import c from "./c"; // Too many dependencies: 3 (max: 2)
```

Examples of **correct** code for this rule:

```javascript
import a from "./a";
import b from "./b"; // Allowed: 2 dependencies (max: 2)
```

## Configuration

This rule accepts a configuration object with the following properties:

### ignoreTypeImports

type: `boolean`

default: `false`

Whether to ignore type imports when counting dependencies.

### max

type: `integer`

default: `10`

Maximum number of dependencies allowed in a module.

## How to use

To **enable** this rule in the CLI or using the config file, you can use:

::: code-group

```bash [CLI]
oxlint --deny import/max-dependencies --import-plugin
```

```json [Config (.oxlintrc.json)]
{
  "plugins": ["import"],
  "rules": {
    "import/max-dependencies": "error"
  }
}
```

:::

## References

- <a v-bind:href="source" target="_blank" rel="noreferrer">Rule Source</a>
