<!-- This file is auto-generated by tasks/website_linter/src/rules/doc_page.rs. Do not edit it manually. -->

<script setup>
import { data } from '../version.data.js';
const source = `https://github.com/oxc-project/oxc/blob/${ data }/crates/oxc_linter/src/rules/import/unambiguous.rs`;
</script>

# import/unambiguous <Badge type="info" text="Restriction" />

<div class="rule-meta">
</div>

### What it does

Warn if a `module` could be mistakenly parsed as a `script` and not pure ESM module

### Why is this bad?

For ESM-only environments helps to determine files that not pure ESM modules

### Examples

Examples of **incorrect** code for this rule:

```js
function x() {}

(function x() { return 42 })()
```

Examples of **correct** code for this rule:

```js
import 'foo'
function x() { return 42 }

export function x() { return 42 }

(function x() { return 42 })()
export {} // simple way to mark side-effects-only file as 'module' without any imports/exports
```

## How to use

To **enable** this rule using the config file or in the CLI, you can use:

::: code-group

```json [Config (.oxlintrc.json)]
{
    "plugins": ["import"],
    "rules": {
        "import/unambiguous": "error"
    }
}
```

```bash [CLI]
oxlint --deny import/unambiguous --import-plugin
```

:::

## References

- <a v-bind:href="source" target="_blank" rel="noreferrer">Rule Source</a>
