<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# oxc/bad-char-at-comparison

<div class="rule-meta">
<span class="default-on">
âœ… This rule is turned on by default.
</span>
</div>

### What it does

This rule warns when the return value of the `charAt` method is used to compare a string of length greater than 1.

### Why is this bad?

The `charAt` method returns a string of length 1. If the return value is compared with a string of length greater than 1, the comparison will always be false.

### Example

```javascript
// Bad: The return value of the `charAt` method is compared with a string of length greater than 1.
a.charAt(4) === "a2";
a.charAt(4) === "/n";

// Good: The return value of the `charAt` method is compared with a string of length 1.
a.charAt(4) === "a";
a.charAt(4) === "\n";
```
