<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# oxc/misrefactored-assign-op

<div class="rule-meta">
</div>

### What it does

https://rust-lang.github.io/rust-clippy/master/#/misrefactored_assign_op

Checks for `a op= a op b` or `a op= b op a` patterns.

### Why is this bad?

Most likely these are bugs where one meant to write `a op= b`.

### Example

```javascript
// Bad
a += a + b;
a -= a - b;

// Good
a += b;
a -= b;
```
