<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

<script setup>
import { data } from '../version.data.js';
const source = `https://github.com/oxc-project/oxc/blob/${ data }/crates/oxc_linter/src/rules/oxc/no_optional_chaining.rs`;
</script>

# oxc/no-optional-chaining <Badge type="info" text="Restriction" />

<div class="rule-meta">
</div>

### What it does

Disallow [optional chaining](https://github.com/tc39/proposal-optional-chaining).

### Why is this bad?

Optional chaining is a relatively new JavaScript feature that may not be
supported in older environments. In some cases, transpiling optional
chaining can result in verbose helper code that impacts bundle size or
performance. This rule is useful when you need to maintain compatibility
with older JavaScript targets or want to avoid the overhead of transpiled
optional chaining.

### Examples

Examples of **incorrect** code for this rule:

```javascript
const foo = obj?.foo;
obj.fn?.();
```

## Configuration

This rule accepts a configuration object with the following properties:

### message

type: `string`

default: `""`

A custom help message to display when optional chaining is found.
For example, "Our output target is ES2016, and optional chaining results in verbose
helpers and should be avoided."

## How to use

To **enable** this rule in the CLI or using the config file, you can use:

::: code-group

```bash [CLI]
oxlint --deny oxc/no-optional-chaining
```

```json [Config (.oxlintrc.json)]
{
  "rules": {
    "oxc/no-optional-chaining": "error"
  }
}
```

:::

## References

- <a v-bind:href="source" target="_blank" rel="noreferrer">Rule Source</a>
