<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

<script setup>
import { data } from '../version.data.js';
const source = `https://github.com/oxc-project/oxc/blob/${ data }/crates/oxc_linter/src/rules/oxc/no_optional_chaining.rs`;
</script>

# oxc/no-optional-chaining <Badge type="info" text="Restriction" />

<div class="rule-meta">
</div>

### What it does

Disallow [optional chaining](https://github.com/tc39/proposal-optional-chaining).

### Examples

Examples of **incorrect** code for this rule:

```javascript
const foo = obj?.foo;
obj.fn?.();
```

### Options

```json
{
  "rules": {
    "no-optional-chaining": [
      "error",
      {
        "message": "Our output target is ES2016, and optional chaining results in verbose
            helpers and should be avoided."
      }
    ]
  }
}
```

- `message`: A custom help message to display when optional chaining is found.

## How to use

To **enable** this rule in the CLI or using the config file, you can use:

::: code-group

```bash [CLI]
oxlint --deny oxc/no-optional-chaining
```

```json [Config (.oxlintrc.json)]
{
  "rules": {
    "oxc/no-optional-chaining": "error"
  }
}
```

:::

## References

- <a v-bind:href="source" target="_blank" rel="noreferrer">Rule Source</a>
