---
title: "oxc/only-used-in-recursion"
category: "Correctness"
default: true
type_aware: false
fix: "fixable_dangerous_fix"
---

<!-- This file is auto-generated by tasks/website_linter/src/rules/doc_page.rs. Do not edit it manually. -->

<script setup>
import { data } from '../version.data.js';
const source = `https://github.com/oxc-project/oxc/blob/${ data }/crates/oxc_linter/src/rules/oxc/only_used_in_recursion.rs`;
</script>

<RuleHeader />

### What it does

Checks for arguments that are only used in recursion with no side-effects.

Inspired by https://rust-lang.github.io/rust-clippy/master/#/only_used_in_recursion

### Why is this bad?

Supplying an argument that is only used in recursive calls is likely a mistake.

It increase cognitive complexity and may impact performance.

### Examples

Examples of **incorrect** code for this rule:

```ts
function test(only_used_in_recursion) {
  return test(only_used_in_recursion);
}
```

Examples of **correct** code for this rule:

```ts
function f(a: number): number {
  if (a == 0) {
    return 1;
  } else {
    return f(a - 1);
  }
}
```

## How to use

To **enable** this rule using the config file or in the CLI, you can use:

::: code-group

```json [Config (.oxlintrc.json)]
{
  "rules": {
    "oxc/only-used-in-recursion": "error"
  }
}
```

```bash [CLI]
oxlint --deny oxc/only-used-in-recursion
```

:::

## References

- <a v-bind:href="source" target="_blank" rel="noreferrer">Rule Source</a>
