<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# oxc/only-used-in-recursion

<div class="rule-meta">
<span class="default-on">
âœ… This rule is turned on by default.
</span>
</div>

### What it does

Checks for arguments that are only used in recursion with no side-effects.

Inspired by https://rust-lang.github.io/rust-clippy/master/#/only_used_in_recursion

### Why is this bad?

Supplying an argument that is only used in recursive calls is likely a mistake.

It increase cognitive complexity and may impact performance.

### Example

```javascript
// Bad - the argument `b` is only used in recursive calls
function f(a: number, b: number): number {
    if a == 0 {
        return 1
    } else {
        return f(a - 1, b + 1)
    }
}

// Good - the argument `b` is omitted
function f(a: number): number {
   if a == 0 {
       return 1
   } else {
       return f(a - 1)
   }
}
```
