<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# oxc/only-used-in-recursion <Badge type="info" text="Correctness" />

<div class="rule-meta">
<Alert class="default-on" type="success">
<span class="emoji">âœ…</span> This rule is turned on by default.
</Alert>
<Alert class="fix" type="info">
<span class="emoji">âš ï¸ğŸ› ï¸ï¸</span> A dangerous auto-fix is available for this rule.
</Alert>
</div>

### What it does

Checks for arguments that are only used in recursion with no side-effects.

Inspired by https://rust-lang.github.io/rust-clippy/master/#/only_used_in_recursion

### Why is this bad?

Supplying an argument that is only used in recursive calls is likely a mistake.

It increase cognitive complexity and may impact performance.

### Example

```ts
// Bad - the argument `b` is only used in recursive calls
function f(a: number, b: number): number {
  if (a == 0) {
    return 1;
  } else {
    return f(a - 1, b + 1);
  }
}

// Good - the argument `b` is omitted
function f(a: number): number {
  if (a == 0) {
    return 1;
  } else {
    return f(a - 1);
  }
}
```

## References

- [Rule Source](https://github.com/oxc-project/oxc/blob/main/crates/oxc_linter/src/rules/oxc/only_used_in_recursion.rs)
