<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# eslint/block-scoped-var <Badge type="info" text="Suspicious" />

<div class="rule-meta">
</div>

### What it does

Generates warnings when variables are used outside of the block in which they were defined.
This emulates C-style block scope.

### Why is this bad?

This rule aims to reduce the usage of variables outside of their binding context
and emulate traditional block scope from other languages.
This is to help newcomers to the language avoid difficult bugs with variable hoisting.

### Examples

Examples of **incorrect** code for this rule:

```js
function doIf() {
  if (true) {
    var build = true;
  }
  console.log(build);
}
```

Examples of **correct** code for this rule:

```js
function doIf() {
  var build;
  if (true) {
    build = true;
  }
  console.log(build);
}
```

## How to use

To **enable** this rule in the CLI or using the config file, you can use:

::: code-group

```bash [CLI]
oxlint --deny block-scoped-var
```

```json [Config (.oxlintrc.json)]
{
  "rules": {
    "block-scoped-var": "error"
  }
}
```

:::

## References

- [Rule Source](https://github.com/oxc-project/oxc/blob/4b4b09e0e44b1e904870920b3dbd2c75398ed9fd/crates/oxc_linter/src/rules/eslint/block_scoped_var.rs)
