<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# eslint/default-param-last <Badge type="info" text="Style" />

<div class="rule-meta">
</div>

### What it does

Enforce default parameters to be last

### Why is this bad?

Putting default parameter at last allows function calls to omit optional tail arguments.

### Example

```javascript
// Correct: optional argument can be omitted
function createUser(id, isAdmin = false) {}
createUser("tabby");

// Incorrect: optional argument can **not** be omitted
function createUser(isAdmin = false, id) {}
createUser(undefined, "tabby");
```

## How to use

To **enable** this rule in the CLI or using the config file, you can use:

::: code-group

```bash [CLI]
oxlint --deny default-param-last
```

```json [Config (.oxlintrc.json)]
{
  "rules": {
    "default-param-last": "error"
  }
}
```

:::

## References

- [Rule Source](https://github.com/oxc-project/oxc/blob/4b4b09e0e44b1e904870920b3dbd2c75398ed9fd/crates/oxc_linter/src/rules/eslint/default_param_last.rs)
