<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

<script setup>
import { data } from '../version.data.js';
const source = `https://github.com/oxc-project/oxc/blob/${ data }/crates/oxc_linter/src/rules/eslint/default_param_last.rs`;
</script>

# eslint/default-param-last <Badge type="info" text="Style" />

<div class="rule-meta">
</div>

### What it does

Enforce default parameters to be last

### Why is this bad?

Putting default parameter at last allows function calls to omit optional tail arguments.

### Examples

Examples of **incorrect** code for this rule:

```javascript
// Incorrect: optional argument can **not** be omitted
function createUser(isAdmin = false, id) {}
createUser(undefined, "tabby");
```

Examples of **correct** code for this rule:

```javascript
function createUser(id, isAdmin = false) {}
createUser("tabby");
```

## How to use

To **enable** this rule in the CLI or using the config file, you can use:

::: code-group

```bash [CLI]
oxlint --deny default-param-last
```

```json [Config (.oxlintrc.json)]
{
  "rules": {
    "default-param-last": "error"
  }
}
```

:::

## References

- <a v-bind:href="source" target="_blank" rel="noreferrer">Rule Source</a>
