<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# eslint/eqeqeq <Badge type="info" text="Pedantic" />

<div class="rule-meta">
<Alert class="fix" type="info">
<span class="emoji">üõ†Ô∏è</span> An auto-fix is available for this rule for some violations.
</Alert>
</div>

### What it does

Requires the use of the `===` and `!==` operators.

### Why is this bad?

Using non-strict equality operators leads to hard to track bugs due to type coercion.

### Examples

Examples of **incorrect** code for this rule:

```js
const a = [];
const b = true;
a == b;
```

The above will evaluate to `true`, but almost surely not want you want.

Examples of **correct** code for this rule:

```js
const a = [];
const b = true;
a === b;
```

The above will evaluate to `false` (an array is not boolean true).

### Options

#### null

```json
  "eslint/eqeqeq": ["error", "always", {"null": "ignore"}]
```

Allow nullish comparison (`foo == null`). The alternative (`foo === null || foo === undefined`) is verbose and has no other benefit.

#### smart

```json
  "eslint/eqeqeq": ["error", "smart"]
```

Allow `==` when comparing:

- the result from `typeof`
- literal values
- nullish

Examples of **incorrect** code for this option:

```js
a == b
[] == true
```

Examples of **correct** code for this option:

```js
typeof foo == "undefined";
"foo" == "bar";
42 == 42;
foo == null;
```

## How to use

To **enable** this rule in the CLI or using the config file, you can use:

::: code-group

```bash [CLI]
oxlint --deny eqeqeq
```

```json [Config (.oxlintrc.json)]
{
  "rules": {
    "eqeqeq": "error"
  }
}
```

:::

## References

- [Rule Source](https://github.com/oxc-project/oxc/blob/f5b53d4cbba0da3f684f1854f5801e8a6d15f214/crates/oxc_linter/src/rules/eslint/eqeqeq.rs)
