<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# eslint/init-declarations <Badge type="info" text="Style" />

<div class="rule-meta">
</div>

### What it does

Require or disallow initialization in variable declarations

### Why is this bad?

In JavaScript, variables can be assigned during declaration, or at any point afterwards using an assignment statement.
For example, in the following code, foo is initialized during declaration, while bar is initialized later.

### Examples

var foo = 1;
var bar;
if (foo) {
bar = 1;
} else {
bar = 2;
}

Examples of incorrect code for the default "always" option:

```js
/*eslint init-declarations: ["error", "always"]*/
function foo() {
  var bar;
  let baz;
}
```

Examples of incorrect code for the "never" option:

```js
/*eslint init-declarations: ["error", "never"]*/
function foo() {
  var bar = 1;
  let baz = 2;
  for (var i = 0; i < 1; i++) {}
}
```

Examples of correct code for the default "always" option:

```js
/*eslint init-declarations: ["error", "always"]*/

function foo() {
  var bar = 1;
  let baz = 2;
  const qux = 3;
}
```

Examples of correct code for the "never" option:

```js
/*eslint init-declarations: ["error", "never"]*/

function foo() {
  var bar;
  let baz;
  const buzz = 1;
}
```

Examples of correct code for the "never", { "ignoreForLoopInit": true } options:

```js
/*eslint init-declarations: ["error", "never", { "ignoreForLoopInit": true }]*/
for (var i = 0; i < 1; i++) {}
```

## How to use

To **enable** this rule in the CLI or using the config file, you can use:

::: code-group

```bash [CLI]
oxlint --deny init-declarations
```

```json [Config (.oxlintrc.json)]
{
  "rules": {
    "init-declarations": "error"
  }
}
```

:::

## References

- [Rule Source](https://github.com/oxc-project/oxc/blob/30318457d425dbf627aa428aad8004f6b92b1c59/crates/oxc_linter/src/rules/eslint/init_declarations.rs)
