<!-- This file is auto-generated by tasks/website_linter/src/rules/doc_page.rs. Do not edit it manually. -->

<script setup>
import { data } from '../version.data.js';
const source = `https://github.com/oxc-project/oxc/blob/${ data }/crates/oxc_linter/src/rules/eslint/init_declarations.rs`;
</script>

# eslint/init-declarations <Badge type="info" text="Style" />

<div class="rule-meta">
</div>

### What it does

Require or disallow initialization in variable declarations

### Why is this bad?

In JavaScript, variables can be assigned during declaration, or at any point afterwards using an assignment statement.
For example, in the following code, foo is initialized during declaration, while bar is initialized later.

### Examples

var foo = 1;
var bar;
if (foo) {
bar = 1;
} else {
bar = 2;
}

Examples of incorrect code for the default "always" option:

```js
/*eslint init-declarations: ["error", "always"]*/
function foo() {
  var bar;
  let baz;
}
```

Examples of incorrect code for the "never" option:

```js
/*eslint init-declarations: ["error", "never"]*/
function foo() {
  var bar = 1;
  let baz = 2;
  for (var i = 0; i < 1; i++) {}
}
```

Examples of correct code for the default "always" option:

```js
/*eslint init-declarations: ["error", "always"]*/

function foo() {
  var bar = 1;
  let baz = 2;
  const qux = 3;
}
```

Examples of correct code for the "never" option:

```js
/*eslint init-declarations: ["error", "never"]*/

function foo() {
  var bar;
  let baz;
  const buzz = 1;
}
```

Examples of correct code for the "never", { "ignoreForLoopInit": true } options:

```js
/*eslint init-declarations: ["error", "never", { "ignoreForLoopInit": true }]*/
for (var i = 0; i < 1; i++) {}
```

## Configuration

This rule accepts a configuration object with the following properties:

### ignoreForLoopInit

type: `boolean`

default: `false`

When set to `true`, allows uninitialized variables in the init expression of `for`, `for-in`, and `for-of` loops.
Only applies when mode is set to `"never"`.

### mode

type: `"always" | "never"`

When set to `"always"` (default), requires that variables be initialized on declaration.
When set to `"never"`, disallows initialization during declaration.

## How to use

To **enable** this rule using the config file or in the CLI, you can use:

::: code-group

```json [Config (.oxlintrc.json)]
{
  "rules": {
    "init-declarations": "error"
  }
}
```

```bash [CLI]
oxlint --deny init-declarations
```

:::

## References

- <a v-bind:href="source" target="_blank" rel="noreferrer">Rule Source</a>
