<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# eslint/no-constructor-return

<div class="rule-meta">
</div>

### What it does

Disallow returning value from constructor

### Why is this bad?

In JavaScript, returning a value in the constructor of a class may be a mistake.
Forbidding this pattern prevents mistakes resulting from unfamiliarity with the language or a copy-paste error.

### Example

Bad:

```rust
class C {
    constructor() { return 42; }
}
```

Good:

```rust
class C {
    constructor() { this.value = 42; }
}
```
