<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# eslint/no-class-assign <Badge type="info" text="Correctness" />

<div class="rule-meta">
<Alert class="default-on" type="success">
<span class="emoji">âœ…</span> This rule is turned on by default.
</Alert>
</div>

### What it does

Disallow reassigning class variables.

### Why is this bad?

`ClassDeclaration` creates a variable that can be re-assigned,
but the re-assignment is a mistake in most cases.

### Example

```javascript
class A {}
A = 123;
let a = new A(); // Error
```

## How to use

To **enable** this rule in the CLI or using the config file, you can use:

::: code-group

```bash [CLI]
oxlint --deny no-class-assign
```

```json [Config (.oxlintrc.json)]
{
  "rules": {
    "no-class-assign": "error"
  }
}
```

:::

## References

- [Rule Source](https://github.com/oxc-project/oxc/blob/30318457d425dbf627aa428aad8004f6b92b1c59/crates/oxc_linter/src/rules/eslint/no_class_assign.rs)
