<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# eslint/no-delete-var <Badge type="info" text="Correctness" />

<div class="rule-meta">
<Alert class="default-on" type="success">
<span class="emoji">âœ…</span> This rule is turned on by default.
</Alert>
</div>

### What it does

The purpose of the delete operator is to remove a property from an object.

### Why is this bad?

Using the delete operator on a variable might lead to unexpected behavior.

### Example

```javascript
var x;
delete x;
```
