<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# eslint/no-empty-function <Badge type="info" text="Restriction" />

<div class="rule-meta">
</div>

### What it does

Disallows the usages of empty functions

### Why is this bad?

Empty functions can reduce readability because readers need to guess whether itâ€™s
intentional or not. So writing a clear comment for empty functions is a good practice.

### Example

Examples of **incorrect** code for this rule:

```javascript
function foo() {}

const bar = () => {};
```

Examples of **correct** code for this rule:

```javascript
function foo() {
  // do nothing
}

function foo() {
  return;
}
const add = (a, b) => a + b;
```

## How to use

To **enable** this rule in the CLI or using the config file, you can use:

::: code-group

```bash [CLI]
oxlint --deny no-empty-function
```

```json [Config (.oxlintrc.json)]
{
  "rules": {
    "no-empty-function": "error"
  }
}
```

:::

## References

- [Rule Source](https://github.com/oxc-project/oxc/blob/bc0670c8a9937f35f43c431ab616950d8f19f8f7/crates/oxc_linter/src/rules/eslint/no_empty_function.rs)
