<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

<script setup>
import { data } from '../version.data.js';
const source = `https://github.com/oxc-project/oxc/blob/${ data }/crates/oxc_linter/src/rules/eslint/no_extra_boolean_cast.rs`;
</script>

# eslint/no-extra-boolean-cast <Badge type="info" text="Correctness" />

<div class="rule-meta">
<Alert class="default-on" type="success">
<span class="emoji">‚úÖ</span> This rule is turned on by default.
</Alert>
<Alert class="fix" type="info">
<span class="emoji">üõ†Ô∏èüí°</span> An auto-fix and a suggestion are available for this rule for some violations.
</Alert>
</div>

### What it does

This rule disallows unnecessary boolean casts.

### Why is this bad?

In contexts such as an if statement's test where the result of the expression will already be coerced to a Boolean,
casting to a Boolean via double negation (!!) or a Boolean call is unnecessary.

### Examples

Examples of **incorrect** code for this rule:

```javascript
var foo = !!!bar;
var foo = Boolean(!!bar);

if (!!foo) {}
if (Boolean(foo)) {}

// with "enforceForLogicalOperands" option enabled
if (!!foo || bar) {}
```

Examples of **correct** code for this rule:

```javascript
var foo = !bar;
var foo = Boolean(bar);

if (foo) {}
if (foo) {}

// with "enforceForLogicalOperands" option enabled
if (foo || bar) {}
```

## How to use

To **enable** this rule in the CLI or using the config file, you can use:

::: code-group

```bash [CLI]
oxlint --deny no-extra-boolean-cast
```

```json [Config (.oxlintrc.json)]
{
  "rules": {
    "no-extra-boolean-cast": "error"
  }
}
```

:::

## References

- <a v-bind:href="source" target="_blank" rel="noreferrer">Rule Source</a>
