<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# eslint/no-global-assign <Badge type="info" text="Correctness" />

<div class="rule-meta">
<Alert class="default-on" type="success">
<span class="emoji">✅</span> This rule is turned on by default.
</Alert>
</div>

### What it does

Disallow modifications to read-only global variables.

### Why is this bad?

In almost all cases, you don’t want to assign a value to these global variables as doing so could result in losing access to important functionality.

### Example

```javascript
Object = null;
```

## References

- [Rule Source](https://github.com/oxc-project/oxc/blob/55744fd9775a032ea83215dadc77a6bff3e2a87d/crates/oxc_linter/src/rules/eslint/no_global_assign.rs)
