<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# eslint/no-unsafe-negation <Badge type="info" text="Correctness" />

<div class="rule-meta">
<Alert class="default-on" type="success">
<span class="emoji">‚úÖ</span> This rule is turned on by default.
</Alert>
<Alert class="fix" type="info">
<span class="emoji">üõ†Ô∏è</span> An auto-fix is available for this rule.
</Alert>
</div>

### What it does

Disallow negating the left operand of relational operators

### Why is this bad?

Just as developers might type -a + b when they mean -(a + b) for the negative of a sum,
they might type !key in object by mistake when they almost certainly mean !(key in object)
to test that a key is not in an object. !obj instanceof Ctor is similar.

### Example

```javascript
if ((!key) in object) {
  //operator precedence makes it equivalent to (!key) in object
  //and type conversion makes it equivalent to (key ? "false" : "true") in object
}
```

## How to use

To **enable** this rule in the CLI or using the config file, you can use:

::: code-group

```bash [CLI]
oxlint --deny no-unsafe-negation
```

```json [Config (.oxlintrc.json)]
{
  "rules": {
    "no-unsafe-negation": "error"
  }
}
```

:::

## References

- [Rule Source](https://github.com/oxc-project/oxc/blob/30318457d425dbf627aa428aad8004f6b92b1c59/crates/oxc_linter/src/rules/eslint/no_unsafe_negation.rs)
