<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# eslint/no-useless-concat <Badge type="info" text="Suspicious" />

<div class="rule-meta">
</div>

### What it does

Disallow unnecessary concatenation of literals or template literals

### Why is this bad?

Itâ€™s unnecessary to concatenate two strings together when they could
be combined into a single literal.

### Examples

Examples of **incorrect** code for this rule:

```javascript
var foo = "a" + "b";
```

```javascript
var foo = "a" + "b" + "c";
```

Examples of **correct** code for this rule:

```javascript
var foo = "a" + bar;
```

// when the string concatenation is multiline

```javascript
var foo = "a" + "b" + "c";
```

## How to use

To **enable** this rule in the CLI or using the config file, you can use:

::: code-group

```bash [CLI]
oxlint --deny no-useless-concat
```

```json [Config (.oxlintrc.json)]
{
  "rules": {
    "no-useless-concat": "error"
  }
}
```

:::

## References

- [Rule Source](https://github.com/oxc-project/oxc/blob/89b6e4c7a880c5e0e6ac98dda359a08759d62e4c/crates/oxc_linter/src/rules/eslint/no_useless_concat.rs)
