<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# eslint/no-unsafe-optional-chaining <Badge type="info" text="Correctness" />

<div class="rule-meta">
<Alert class="default-on" type="success">
<span class="emoji">âœ…</span> This rule is turned on by default.
</Alert>
</div>

### What it does

Disallow use of optional chaining in contexts where the undefined value is not allowed

### Why is this bad?

The optional chaining (?.) expression can short-circuit with a return value of undefined.
Therefore, treating an evaluated optional chaining expression as a function, object, number, etc.,
can cause TypeError or unexpected results. For example:

### Example

```javascript
var obj = undefined;
1 in obj?.foo; // TypeError
with (obj?.foo); // TypeError
for (bar of obj?.foo); // TypeError
bar instanceof obj?.foo; // TypeError
const { bar } = obj?.foo; // TypeError
```

## How to use

To **enable** this rule in the CLI or using the config file, you can use:

::: code-group

```bash [CLI]
oxlint --deny no-unsafe-optional-chaining
```

```json [Config (.oxlintrc.json)]
{
  "rules": {
    "no-unsafe-optional-chaining": "error"
  }
}
```

:::

## References

- [Rule Source](https://github.com/oxc-project/oxc/blob/30318457d425dbf627aa428aad8004f6b92b1c59/crates/oxc_linter/src/rules/eslint/no_unsafe_optional_chaining.rs)
