<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# eslint/no-useless-rename

<div class="rule-meta">
<Alert class="default-on" type="success">
<span class="emoji">âœ…</span> This rule is turned on by default.
</Alert>
</div>

### What it does

Disallow renaming import, export, and destructured assignments to the same name.

### Why is this bad?

It is unnecessary to rename a variable to the same name.

### Example

```javascript
// Bad
import { foo as foo } from "foo";
const { bar: bar } = obj;
export { baz as baz };

// Good
import { foo } from "foo";
const { bar: renamed } = obj;
export { baz };
```
