<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

<script setup>
import { data } from '../version.data.js';
const source = `https://github.com/oxc-project/oxc/blob/${ data }/crates/oxc_linter/src/rules/eslint/no_with.rs`;
</script>
# eslint/no-with <Badge type="info" text="Correctness" />
<div class="rule-meta">
<Alert class="default-on" type="success">
<span class="emoji">âœ…</span> This rule is turned on by default.
</Alert>
</div>

### What it does

Disallow `with` statements

### Why is this bad?

The with statement is potentially problematic because it adds members of an object to the current scope, making it impossible to tell what a variable inside the block actually refers to.

### Examples

Examples of **incorrect** code for this rule:
```javascript
with (point) {
    r = Math.sqrt(x * x + y * y); // is r a member of point?
}
```


## How to use

To **enable** this rule in the CLI or using the config file, you can use:

::: code-group

```bash [CLI]
oxlint --deny no-with
```

```json [Config (.oxlintrc.json)]
{
    "rules": {
        "no-with": "error"
    }
}
```

:::


## References

- <a v-bind:href="source" target="_blank" rel="noreferrer">Rule Source</a>
