<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

<script setup>
import { data } from '../version.data.js';
const source = `https://github.com/oxc-project/oxc/blob/${ data }/crates/oxc_linter/src/rules/eslint/prefer_exponentiation_operator.rs`;
</script>

# eslint/prefer-exponentiation-operator <Badge type="info" text="Style" />

<div class="rule-meta">
</div>

### What it does

Disallow the use of Math.pow in favor of the ** operator

### Why is this bad?

Introduced in ES2016, the infix exponentiation operator ** is an alternative for the
standard Math.pow function. Infix notation is considered to be more readable and thus more
preferable than the function notation.

### Examples

Examples of **incorrect** code for this rule:

```javascript
Math.pow(a, b);
```

Examples of **correct** code for this rule:

```javascript
a ** b;
```

## How to use

To **enable** this rule in the CLI or using the config file, you can use:

::: code-group

```bash [CLI]
oxlint --deny prefer-exponentiation-operator
```

```json [Config (.oxlintrc.json)]
{
  "rules": {
    "prefer-exponentiation-operator": "error"
  }
}
```

:::

## References

- <a v-bind:href="source" target="_blank" rel="noreferrer">Rule Source</a>
