---
title: "eslint/prefer-numeric-literals"
category: "Style"
default: false
type_aware: false
fix: "conditional_fix"
---

<!-- This file is auto-generated by tasks/website_linter/src/rules/doc_page.rs. Do not edit it manually. -->

<script setup>
import { data } from '../version.data.js';
const source = `https://github.com/oxc-project/oxc/blob/${ data }/crates/oxc_linter/src/rules/eslint/prefer_numeric_literals.rs`;
</script>

<RuleHeader />

### What it does

Disallow parseInt() and Number.parseInt() in favor of binary, octal, and hexadecimal
literals.

### Why is this bad?

The parseInt() and Number.parseInt() functions can be used to turn binary, octal, and
hexadecimal strings into integers. As binary, octal, and hexadecimal literals are supported
in ES2015, this rule encourages use of those numeric literals instead of parseInt() or
Number.parseInt().

### Examples

Examples of **incorrect** code for this rule:

```javascript
parseInt("111110111", 2) === 503;
parseInt(`111110111`, 2) === 503;
parseInt("767", 8) === 503;
parseInt("1F7", 16) === 503;
Number.parseInt("111110111", 2) === 503;
Number.parseInt("767", 8) === 503;
Number.parseInt("1F7", 16) === 503;
```

## How to use

To **enable** this rule using the config file or in the CLI, you can use:

::: code-group

```json [Config (.oxlintrc.json)]
{
  "rules": {
    "prefer-numeric-literals": "error"
  }
}
```

```bash [CLI]
oxlint --deny prefer-numeric-literals
```

:::

## References

- <a v-bind:href="source" target="_blank" rel="noreferrer">Rule Source</a>
