---
title: "eslint/radix"
category: "Pedantic"
default: false
type_aware: false
fix: "conditional_dangerous_fix"
---

<!-- This file is auto-generated by tasks/website_linter/src/rules/doc_page.rs. Do not edit it manually. -->

<script setup>
import { data } from '../version.data.js';
const source = `https://github.com/oxc-project/oxc/blob/${ data }/crates/oxc_linter/src/rules/eslint/radix.rs`;
</script>

<RuleHeader />

### What it does

Enforce the consistent use of the radix argument when using `parseInt()`.

### Why is this bad?

Using the `parseInt()` function without specifying
the radix can lead to unexpected results.

See the
[MDN documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt#radix)
for more information.

### Examples

Examples of **incorrect** code for this rule:

```javascript
var num = parseInt("071"); // 57
```

Examples of **correct** code for this rule:

```javascript
var num = parseInt("071", 10); // 71
```

## Configuration

This rule accepts one of the following string values:

### `"always"`

Always require the radix parameter when using `parseInt()`.

### `"as-needed"`

Only require the radix parameter when necessary.

## How to use

To **enable** this rule using the config file or in the CLI, you can use:

::: code-group

```json [Config (.oxlintrc.json)]
{
  "rules": {
    "radix": "error"
  }
}
```

```bash [CLI]
oxlint --deny radix
```

:::

## References

- <a v-bind:href="source" target="_blank" rel="noreferrer">Rule Source</a>
