<!-- This file is auto-generated by tasks/website_linter/src/rules/doc_page.rs. Do not edit it manually. -->

<script setup>
import { data } from '../version.data.js';
const source = `https://github.com/oxc-project/oxc/blob/${ data }/crates/oxc_linter/src/rules/eslint/radix.rs`;
</script>

# eslint/radix <Badge type="info" text="Pedantic" />

<div class="rule-meta">
<Alert class="fix" type="info">
<span class="emoji">‚ö†Ô∏èüõ†Ô∏èÔ∏è</span> A dangerous auto-fix is available for this rule for some violations.
</Alert>
</div>

### What it does

Enforce the consistent use of the radix argument when using `parseInt()`.

### Why is this bad?

Using the `parseInt()` function without specifying
the radix can lead to unexpected results.

See the
[MDN documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt#radix)
for more information.

### Examples

Examples of **incorrect** code for this rule:

```javascript
var num = parseInt("071"); // 57
```

Examples of **correct** code for this rule:

```javascript
var num = parseInt("071", 10); // 71
```

## Configuration

This rule accepts one of the following string values:

### `"always"`

Always require the radix parameter when using `parseInt()`.

### `"as-needed"`

Only require the radix parameter when necessary.

## How to use

To **enable** this rule in the CLI or using the config file, you can use:

::: code-group

```bash [CLI]
oxlint --deny radix
```

```json [Config (.oxlintrc.json)]
{
  "rules": {
    "radix": "error"
  }
}
```

:::

## References

- <a v-bind:href="source" target="_blank" rel="noreferrer">Rule Source</a>
