<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# eslint/use-isnan <Badge type="info" text="Correctness" />

<div class="rule-meta">
<Alert class="default-on" type="success">
<span class="emoji">‚úÖ</span> This rule is turned on by default.
</Alert>
<Alert class="fix" type="info">
<span class="emoji">üõ†Ô∏è</span> An auto-fix is available for this rule for some violations.
</Alert>
</div>

### What it does

Disallows checking against NaN without using isNaN() call.

### Why is this bad?

In JavaScript, NaN is a special value of the Number type.
It‚Äôs used to represent any of the ‚Äúnot-a-number‚Äù values represented
by the double-precision 64-bit format as specified by the IEEE Standard
for Binary Floating-Point Arithmetic.

Because NaN is unique in JavaScript by not being equal to anything, including itself,
the results of comparisons to NaN are confusing:

- NaN === NaN or NaN == NaN evaluate to false
- NaN !== NaN or NaN != NaN evaluate to true

Therefore, use Number.isNaN() or global isNaN() functions to test whether a value is NaN.

### Example

```javascript
foo == NaN;
foo === NaN;
foo <= NaN;
foo > NaN;
```

## How to use

To **enable** this rule in the CLI or using the config file, you can use:

::: code-group

```bash [CLI]
oxlint --deny use-isnan
```

```json [Config (.oxlintrc.json)]
{
  "rules": {
    "use-isnan": "error"
  }
}
```

:::

## References

- [Rule Source](https://github.com/oxc-project/oxc/blob/aba3654b81166a7d52f13ac067ffa4bab5e702c9/crates/oxc_linter/src/rules/eslint/use_isnan.rs)
