<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# node/no-new-require <Badge type="info" text="Restriction" />

<div class="rule-meta">
</div>

### What it does

Warn about calling `new` on `require`.

### Why is this bad?

The `require` function is used to include modules and might return a constructor. As this
is not always the case this can be confusing.

### Examples

Examples of **incorrect** code for this rule:

```js
var appHeader = new require("app-header");
```

Examples of **correct** code for this rule:

```js
var AppHeader = require("app-header");
var appHeader = new AppHeader();
```

## References

- [Rule Source](https://github.com/oxc-project/oxc/blob/5172f324579e6a3858a816912e6d8a284d5debad/crates/oxc_linter/src/rules/node/no_new_require.rs)
