---
title: "jest/no-jasmine-globals"
category: "Style"
default: false
type_aware: false
fix: "conditional_fix"
---

<!-- This file is auto-generated by tasks/website_linter/src/rules/doc_page.rs. Do not edit it manually. -->

<script setup>
import { data } from '../version.data.js';
const source = `https://github.com/oxc-project/oxc/blob/${ data }/crates/oxc_linter/src/rules/jest/no_jasmine_globals.rs`;
</script>

<RuleHeader />

### What it does

This rule reports on any usage of Jasmine globals, which is not ported to
Jest, and suggests alternatives from Jest's own API.

### Why is this bad?

When migrating from Jasmine to Jest, relying on Jasmine-specific globals
creates compatibility issues and prevents taking advantage of Jest's
improved testing features and better error reporting.

### Examples

Examples of **incorrect** code for this rule:

```javascript
jasmine.DEFAULT_TIMEOUT_INTERVAL = 5000;
test("my test", () => {
  pending();
});
test("my test", () => {
  jasmine.createSpy();
});
```

Examples of **correct** code for this rule:

```javascript
jest.setTimeout(5000);
test("my test", () => {
  // Use test.skip() instead of pending()
});
test.skip("my test", () => {
  // Skipped test
});
test("my test", () => {
  jest.fn(); // Use jest.fn() instead of jasmine.createSpy()
});
```

## How to use

To **enable** this rule using the config file or in the CLI, you can use:

::: code-group

```json [Config (.oxlintrc.json)]
{
  "plugins": ["jest"],
  "rules": {
    "jest/no-jasmine-globals": "error"
  }
}
```

```bash [CLI]
oxlint --deny jest/no-jasmine-globals --jest-plugin
```

:::

## References

- <a v-bind:href="source" target="_blank" rel="noreferrer">Rule Source</a>
