<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# jest/prefer-todo

<div class="rule-meta">
<span class="fix">
ğŸ› ï¸ An auto-fix is available for this rule.
</span>
</div>

### What it does
When test cases are empty then it is better to mark them as `test.todo` as it
will be highlighted in the summary output.

### Why is this bad?

This rule triggers a warning if empty test cases are used without 'test.todo'.

### Example

```javascript
test('i need to write this test'); // invalid
test('i need to write this test', () => {}); // invalid
test.skip('i need to write this test', () => {}); // invalid

test.todo('i need to write this test');
```

