<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# jsx_a11y/alt-text <Badge type="info" text="Correctness" />

<div class="rule-meta">
<Alert class="default-on" type="success">
<span class="emoji">âœ…</span> This rule is turned on by default.
</Alert>
</div>

### What it does

Enforce that all elements that require alternative text have meaningful
information to relay back to the end user.

### Why is this necessary?

Alternative text is a critical component of accessibility for screen
reader users, enabling them to understand the content and function
of an element.

### What it checks

This rule checks for alternative text on the following elements:
`<img>`, `<area>`, `<input type="image">`, and `<object>`.

### How to fix it

Ensure that the `alt` attribute is present and contains meaningful
text that describes the element's content or purpose.

### Example

Examples of **incorrect** code for this rule:

```jsx
<img src="flower.jpg" alt="A close-up of a white daisy" />
```

Examples of **correct** code for this rule:

```jsx
<img src="flower.jpg" />
```

## How to use

To **enable** this rule in the CLI or using the config file, you can use:

::: code-group

```bash [CLI]
oxlint --deny jsx-a11y/alt-text --jsx-a11y-plugin
```

```json [Config (.oxlintrc.json)]
{
  "plugins": ["jsx-a11y"],
  "rules": {
    "jsx-a11y/alt-text": "error"
  }
}
```

:::

## References

- [Rule Source](https://github.com/oxc-project/oxc/blob/f5b53d4cbba0da3f684f1854f5801e8a6d15f214/crates/oxc_linter/src/rules/jsx_a11y/alt_text.rs)
