<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# jsx_a11y/click-events-have-key-events

<div class="rule-meta">
</div>

### What it does

Enforce onClick is accompanied by at least one of the following: onKeyUp, onKeyDown, onKeyPress.

### Why is this bad?

Coding for the keyboard is important for users with physical disabilities who cannot use a mouse, AT compatibility, and screenreader users.
This does not apply for interactive or hidden elements.

### Example

```jsx
// Good
<div onClick={() => void 0} onKeyDown={() => void 0} />

// Bad
<div onClick={() => void 0} />
```
