<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# jsx_a11y/HTML-has-lang

<div class="rule-meta">
</div>

### What it does

Ensures that every HTML document has a lang attribute

### Why is this bad?

If the language of a web page is not specified,
the screen reader assumes the default language set by the user.
Language settings become an issue for users who speak multiple languages
and access website in more than one language.

### Example

```javascript
// Bad
<html />

// Good
<html lang="en" />
```
