<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# jsx_a11y/lang <Badge type="info" text="Correctness" />

<div class="rule-meta">
</div>

### What it does

The lang prop on the `<html>` element must be a valid IETF's BCP 47 language tag.

### Why is this bad?

If the language of a webpage is not specified as valid,
the screen reader assumes the default language set by the user.
Language settings become an issue for users who speak multiple languages
and access website in more than one language.

### Examples

Examples of **incorrect** code for this rule:

```jsx
<html>
<html lang="foo">
```

Examples of **correct** code for this rule:

```jsx
<html lang="en">
<html lang="en-US">
```

### Resources

- [eslint-plugin-jsx-a11y/lang](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/v6.9.0/docs/rules/lang.md)
- [IANA Language Subtag Registry](https://www.iana.org/assignments/language-subtag-registry/language-subtag-registry)

## References

- [Rule Source](https://github.com/oxc-project/oxc/blob/36ebb3e7841818c238c44349d6cf859db4177d55/crates/oxc_linter/src/rules/jsx_a11y/lang.rs)
