<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# jsx_a11y/no-aria-hidden-on-focusable

<div class="rule-meta">
</div>

### What it does
Enforces that `aria-hidden="true"` is not set on focusable elements.

### Why is this bad?
`aria-hidden="true"` on focusable elements can lead to confusion or unexpected behavior for screen reader users.

### Example
```javascript
// Bad
<div aria-hidden="true" tabIndex="0" />

// Good
<div aria-hidden="true" />
```

