<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

<script setup>
import { data } from '../version.data.js';
const source = `https://github.com/oxc-project/oxc/blob/${ data }/crates/oxc_linter/src/rules/jsx_a11y/no_redundant_roles.rs`;
</script>

# jsx_a11y/no-redundant-roles <Badge type="info" text="Correctness" />

<div class="rule-meta">
<Alert class="fix" type="info">
<span class="emoji">üõ†Ô∏è</span> An auto-fix is available for this rule.
</Alert>
</div>

### What it does

Enforces that the explicit `role` property is not the same as
implicit/default role property on element.

### Why is this bad?

Redundant roles can lead to confusion and verbosity in the codebase.

### Examples

Examples of **incorrect** code for this rule:

```jsx
<nav role="navigation" />;
```

Examples of **correct** code for this rule:

```jsx
<nav />;
```

## How to use

To **enable** this rule in the CLI or using the config file, you can use:

::: code-group

```bash [CLI]
oxlint --deny jsx-a11y/no-redundant-roles --jsx-a11y-plugin
```

```json [Config (.oxlintrc.json)]
{
  "plugins": ["jsx-a11y"],
  "rules": {
    "jsx-a11y/no-redundant-roles": "error"
  }
}
```

:::

## References

- <a v-bind:href="source" target="_blank" rel="noreferrer">Rule Source</a>
