<!-- This file is auto-generated by tasks/website_linter/src/rules/doc_page.rs. Do not edit it manually. -->

<script setup>
import { data } from '../version.data.js';
const source = `https://github.com/oxc-project/oxc/blob/${ data }/crates/oxc_linter/src/rules/jsx_a11y/role_has_required_aria_props.rs`;
</script>

# jsx_a11y/role-has-required-aria-props <Badge type="info" text="Correctness" />

<div class="rule-meta">
</div>

### What it does

Enforces that elements with ARIA roles must have all required attributes
for that role.

### Why is this bad?

Certain ARIA roles require specific attributes to express necessary
semantics for assistive technology.

### Examples

Examples of **incorrect** code for this rule:

```jsx
<div role="checkbox" />
```

Examples of **correct** code for this rule:

```jsx
<div role="checkbox" aria-checked="false" />
```

## How to use

To **enable** this rule using the config file or in the CLI, you can use:

::: code-group

```json [Config (.oxlintrc.json)]
{
    "plugins": ["jsx-a11y"],
    "rules": {
        "jsx-a11y/role-has-required-aria-props": "error"
    }
}
```

```bash [CLI]
oxlint --deny jsx-a11y/role-has-required-aria-props --jsx-a11y-plugin
```

:::

## References

- <a v-bind:href="source" target="_blank" rel="noreferrer">Rule Source</a>
