<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# jsx_a11y/tabindex-no-positive

<div class="rule-meta">
</div>

### What it does

Enforces that positive values for the tabIndex attribute are not used in JSX.

### Why is this bad?

Using tabIndex values greater than 0 can make navigation and interaction difficult for keyboard and assistive technology users, disrupting the logical order of content.

### Example

```javascript
// Bad
<span tabIndex="1">foo</span>

// Good
<span tabIndex="0">foo</span>
<span tabIndex="-1">bar</span>
```
